<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Experimental design concepts for label-free proteomics experiments</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">hupo21</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="sourcesOfVariability.html">Sources of Variability</a>
    </li>
    <li>
      <a href="experimentalDesign.html">Experimental Design</a>
    </li>
    <li>
      <a href="msqrob2guiTutorial.html">msqrob2 tutorial</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/hupo21">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Experimental design concepts for label-free proteomics experiments</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date"><a href="https://statomics.github.io">statOmics</a>, Ghent University</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<p>This is part of the online course <a href="https://statomics.github.io/hupo21/">Experimental Design and Data-Analysis in Label-Free Quantitative LC/MS Proteomics - A Tutorial with msqrob2 (hupo21)</a></p>
<details>
<summary> Click to see libraries that are loaded </summary>
<p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">library</span>(limma)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw">library</span>(QFeatures)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="kw">library</span>(msqrob2)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="kw">library</span>(plotly)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="kw">library</span>(gridExtra)</span></code></pre></div>
</p>
</details>
<div id="overview" class="section level1">
<h1><span class="header-section-number">1</span> Overview</h1>
<ul>
<li>Sample size</li>
<li>Randomized Complete Block Designs</li>
</ul>
</div>
<div id="sample-size" class="section level1">
<h1><span class="header-section-number">2</span> Sample size</h1>
<div id="statistical-models" class="section level2">
<h2><span class="header-section-number">2.1</span> Statistical models</h2>
<ul>
<li>Upon preprocessing and summarization we model the log2 transformed protein expression (<span class="math inline">\(y_{i}\)</span>) values using a linear model</li>
</ul>
<p><span class="math display">\[y_{i}= \beta_0 + \beta_1 x_{i,1} + \beta x_{i,2} + ... + \epsilon_i\]</span></p>
<p>with</p>
<ul>
<li><span class="math inline">\(\beta_0\)</span> the intercept</li>
<li><span class="math inline">\(\beta_j\)</span> the slope for predictor <span class="math inline">\(x_{i,j}\)</span></li>
<li><span class="math inline">\(x_{i,j}\)</span> a continuous predictor (such as age) or a dummy variable that can take values of 0 and 1 and that will used for factors (e.g treatment: cancer, normal).</li>
</ul>
<p>Example:</p>
<ul>
<li>Estrogen Receptor Positive Breast cancer tissues from from patients treated with tamoxifen upon recurrence have been assessed in a proteomics study.</li>
<li>Half of the patients had a good outcome (or) and the other half had a poor outcome (pd).</li>
<li>The proteomes have been assessed using an LTQ-Orbitrap and the thermo output .RAW files were searched with MaxQuant (version 1.4.1.2) against the human proteome database (FASTA version 2012-09, human canonical proteome).</li>
</ul>
<p>We model the data for a single protein using a model:</p>
<p><span class="math display">\[y_{i}= \beta_0 + \beta_{PD} x_{i,PD} + \epsilon_i\]</span></p>
<p>with <span class="math inline">\(x_{i,PD}=\begin{cases} 0&amp; \text{good outcome}\\ 1&amp; \text{poor outcome} \end{cases}\)</span>.</p>
<div id="read-data" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Read data</h3>
<details>
<summary> Click to see background and code </summary>
<p>
<ol style="list-style-type: decimal">
<li>We use a peptides.txt file from MS-data quantified with maxquant that contains MS1 intensities summarized at the peptide level.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>peptidesFile &lt;-<span class="st"> &quot;https://raw.githubusercontent.com/statOmics/pda21/data/quantification/cancer/peptides3vs3.txt&quot;</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below:</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>ecols &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;Intensity</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="kw">names</span>(<span class="kw">read.delim</span>(peptidesFile)))</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Read the data and store it in QFeatures object</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>pe &lt;-<span class="st"> </span><span class="kw">readQFeatures</span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="dt">table =</span> peptidesFile,</span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="dt">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="dt">ecol =</span> ecols,</span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="dt">name =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="design" class="section level3">
<h3><span class="header-section-number">2.1.2</span> Design</h3>
<details>
<summary> Click to see background and code </summary>
<p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>pe <span class="op">%&gt;%</span><span class="st"> </span>colnames</span></code></pre></div>
<pre><code>## CharacterList of length 1
## [[&quot;peptideRaw&quot;]] Intensity.OR.01 Intensity.OR.04 ... Intensity.PD.04</code></pre>
<ul>
<li>Note, that the sample names the outcome.</li>
<li>We update the colData with information on the design</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw">colData</span>(pe)<span class="op">$</span>prognosis &lt;-<span class="st">  </span>pe[[<span class="st">&quot;peptideRaw&quot;</span>]] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="st">  </span>colnames <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="st">  </span><span class="kw">substr</span>(<span class="dv">11</span>,<span class="dv">12</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="st">  </span>as.factor</span></code></pre></div>
<ul>
<li>We explore the colData</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="kw">colData</span>(pe)</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 1 column
##                 prognosis
##                  &lt;factor&gt;
## Intensity.OR.01        OR
## Intensity.OR.04        OR
## Intensity.OR.07        OR
## Intensity.PD.02        PD
## Intensity.PD.03        PD
## Intensity.PD.04        PD</code></pre>
</p>
</details>
</div>
<div id="preprocessing" class="section level3">
<h3><span class="header-section-number">2.1.3</span> Preprocessing</h3>
<details>
<summary> Click to see code for preprocessing </summary>
<p>
<ol style="list-style-type: decimal">
<li>Log-transform</li>
</ol>
<ul>
<li>We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="op">$</span>nNonZero &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</span></code></pre></div>
<ul>
<li>Peptides with zero intensities are missing peptides and should be represent with a <code>NA</code> value rather than <code>0</code>.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>pe &lt;-<span class="st"> </span><span class="kw">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<ul>
<li>Logtransform data with base 2</li>
</ul>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>pe &lt;-<span class="st"> </span><span class="kw">logTransform</span>(pe, <span class="dt">base =</span> <span class="dv">2</span>, <span class="dt">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Filtering</li>
</ol>
<details>
<summary> Click to see code to filter the data </summary>
<p>
<ul>
<li>Handling overlapping protein groups</li>
</ul>
<p>In our approach a peptide can map to multiple proteins, as long as there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe, <span class="op">~</span><span class="st"> </span>Proteins <span class="op">%in%</span><span class="st"> </span><span class="kw">smallestUniqueGroups</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>Proteins))</span></code></pre></div>
<ul>
<li>Remove reverse sequences (decoys) and contaminants</li>
</ul>
<p>We now remove the contaminants, peptides that map to decoy sequences, and proteins which were only identified by peptides with modifications.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span>Reverse <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span>
<span id="cb14-2"><a href="#cb14-2"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span><span class="st"> </span>Contaminant <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span></code></pre></div>
<ul>
<li>Drop peptides that were only identified in one sample</li>
</ul>
<p>We keep peptides that were observed at last twice.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span><span class="st"> </span>nNonZero <span class="op">&gt;=</span><span class="dv">2</span>)</span>
<span id="cb15-2"><a href="#cb15-2"></a><span class="kw">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 22413</code></pre>
<p>We keep 22413 peptides upon filtering.</p>
<ol start="3" style="list-style-type: decimal">
<li>Normalization</li>
</ol>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>pe &lt;-<span class="st"> </span><span class="kw">normalize</span>(pe, </span>
<span id="cb17-2"><a href="#cb17-2"></a>                <span class="dt">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb17-3"><a href="#cb17-3"></a>                <span class="dt">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb17-4"><a href="#cb17-4"></a>                <span class="dt">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Summarization</li>
</ol>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>pe &lt;-<span class="st"> </span><span class="kw">aggregateFeatures</span>(pe,</span>
<span id="cb18-2"><a href="#cb18-2"></a>  <span class="dt">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb18-3"><a href="#cb18-3"></a>  <span class="dt">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb18-4"><a href="#cb18-4"></a>  <span class="dt">na.rm =</span> <span class="ot">TRUE</span>,</span>
<span id="cb18-5"><a href="#cb18-5"></a>  <span class="dt">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
</div>
<div id="effect-size" class="section level2">
<h2><span class="header-section-number">2.2</span> Effect size?</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="kw">library</span>(ExploreModelMatrix)</span>
<span id="cb19-2"><a href="#cb19-2"></a><span class="kw">VisualizeDesign</span>(<span class="kw">colData</span>(pe),<span class="dt">designFormula =</span> <span class="op">~</span>prognosis)<span class="op">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<p><span class="math display">\[
\begin{array}{rclcl}
E[Y|OR] &amp;=&amp; \beta_0\\
E[Y|PD] &amp;=&amp; \beta_0 + \beta_{PD}\\
\log_2 FC_{PD-OR} &amp;=&amp;  \beta_0 + \beta_{PD} -  \beta_0 &amp;=&amp;\beta_{PD}
\end{array}
\]</span></p>
</div>
<div id="statistical-inference" class="section level2">
<h2><span class="header-section-number">2.3</span> Statistical Inference</h2>
<p>We want to find proteins that are differential abundant</p>
<p><span class="math inline">\(\rightarrow\)</span> use a statistical test</p>
<ul>
<li><span class="math inline">\(H_0\)</span>: <span class="math inline">\(\log_2 FC_{PD-OR} = 0\)</span> or <span class="math inline">\(\beta_{PD} = 0\)</span></li>
<li><span class="math inline">\(H_1\)</span>: <span class="math inline">\(\log_2 FC_{PD-OR} \neq 0\)</span> or <span class="math inline">\(\beta_{PD} \neq 0\)</span></li>
</ul>
</div>
<div id="experiment-with-3-vs-3-comparison" class="section level2">
<h2><span class="header-section-number">2.4</span> Experiment with 3 vs 3 comparison</h2>
<details>
<summary> Click to see code to inference </summary>
<p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>pe &lt;-<span class="st"> </span><span class="kw">msqrob</span>(<span class="dt">object =</span> pe, <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="dt">formula =</span> <span class="op">~</span>prognosis)</span>
<span id="cb21-2"><a href="#cb21-2"></a>L &lt;-<span class="st"> </span><span class="kw">makeContrast</span>(<span class="st">&quot;prognosisPD=0&quot;</span>, <span class="dt">parameterNames =</span> <span class="kw">c</span>(<span class="st">&quot;prognosisPD&quot;</span>))</span>
<span id="cb21-3"><a href="#cb21-3"></a>pe &lt;-<span class="st"> </span><span class="kw">hypothesisTest</span>(<span class="dt">object =</span> pe, <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="dt">contrast =</span> L)</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>volcano3x3 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>prognosisPD <span class="op">%&gt;%</span><span class="st"> </span>na.exclude,</span>
<span id="cb22-2"><a href="#cb22-2"></a>                  <span class="kw">aes</span>(<span class="dt">x =</span> logFC, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(pval), <span class="dt">color =</span> adjPval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)) <span class="op">+</span></span>
<span id="cb22-3"><a href="#cb22-3"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">cex =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb22-4"><a href="#cb22-4"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">alpha</span>(<span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb22-5"><a href="#cb22-5"></a><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></span>
<span id="cb22-6"><a href="#cb22-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="kw">sum</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>prognosisPD<span class="op">$</span>adjPval<span class="op">&lt;</span><span class="fl">0.05</span>,<span class="dt">na.rm=</span><span class="ot">TRUE</span>),<span class="st">&quot; proteins are found to be DA&quot;</span>))</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>volcano3x3</span></code></pre></div>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p>Upon correction for multiple testing with using the false discovery rate (FDR) method no proteins are found to be differentially expressed.</p>
<p>Note, that you can refresh the concept of multiple testing and FDR [here] (<a href="https://statomics.github.io/PDA21/pda_quantification_inference.html#14_Multiple_hypothesis_testing" class="uri">https://statomics.github.io/PDA21/pda_quantification_inference.html#14_Multiple_hypothesis_testing</a>).</p>
</div>
<div id="experiment-with-a-6-vs-6-comparison" class="section level2">
<h2><span class="header-section-number">2.5</span> Experiment with a 6 vs 6 comparison</h2>
<div id="read-data-1" class="section level3">
<h3><span class="header-section-number">2.5.1</span> Read data</h3>
<details>
<summary> Click to see background and code </summary>
<p>
<ol style="list-style-type: decimal">
<li>We use a peptides.txt file from MS-data quantified with maxquant that contains MS1 intensities summarized at the peptide level.</li>
</ol>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>peptidesFile &lt;-<span class="st"> &quot;https://raw.githubusercontent.com/statOmics/pda21/data/quantification/cancer/peptides6vs6.txt&quot;</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below:</li>
</ol>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>ecols &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;Intensity</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="kw">names</span>(<span class="kw">read.delim</span>(peptidesFile)))</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Read the data and store it in QFeatures object</li>
</ol>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>pe &lt;-<span class="st"> </span><span class="kw">readQFeatures</span>(</span>
<span id="cb26-2"><a href="#cb26-2"></a>  <span class="dt">table =</span> peptidesFile,</span>
<span id="cb26-3"><a href="#cb26-3"></a>  <span class="dt">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb26-4"><a href="#cb26-4"></a>  <span class="dt">ecol =</span> ecols,</span>
<span id="cb26-5"><a href="#cb26-5"></a>  <span class="dt">name =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="design-1" class="section level3">
<h3><span class="header-section-number">2.5.2</span> Design</h3>
<details>
<summary> Click to see background and code </summary>
<p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>pe <span class="op">%&gt;%</span><span class="st"> </span>colnames</span></code></pre></div>
<pre><code>## CharacterList of length 1
## [[&quot;peptideRaw&quot;]] Intensity.OR.01 Intensity.OR.04 ... Intensity.PD.08</code></pre>
<ul>
<li>Note, that the sample names the outcome.</li>
<li>We update the colData with information on the design</li>
</ul>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="kw">colData</span>(pe)<span class="op">$</span>prognosis &lt;-<span class="st">  </span>pe[[<span class="st">&quot;peptideRaw&quot;</span>]] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb29-2"><a href="#cb29-2"></a><span class="st">  </span>colnames <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb29-3"><a href="#cb29-3"></a><span class="st">  </span><span class="kw">substr</span>(<span class="dv">11</span>,<span class="dv">12</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb29-4"><a href="#cb29-4"></a><span class="st">  </span>as.factor</span></code></pre></div>
<ul>
<li>We explore the colData</li>
</ul>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="kw">colData</span>(pe)</span></code></pre></div>
<pre><code>## DataFrame with 12 rows and 1 column
##                 prognosis
##                  &lt;factor&gt;
## Intensity.OR.01        OR
## Intensity.OR.04        OR
## Intensity.OR.07        OR
## Intensity.OR.09        OR
## Intensity.OR.10        OR
## ...                   ...
## Intensity.PD.03        PD
## Intensity.PD.04        PD
## Intensity.PD.06        PD
## Intensity.PD.07        PD
## Intensity.PD.08        PD</code></pre>
</p>
</details>
</div>
<div id="preprocessing-1" class="section level3">
<h3><span class="header-section-number">2.5.3</span> Preprocessing</h3>
<details>
<summary> Click to see code for preprocessing </summary>
<p>
<ol style="list-style-type: decimal">
<li>Log-transform</li>
</ol>
<ul>
<li>We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.</li>
</ul>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a><span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="op">$</span>nNonZero &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</span></code></pre></div>
<ul>
<li>Peptides with zero intensities are missing peptides and should be represent with a <code>NA</code> value rather than <code>0</code>.</li>
</ul>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a>pe &lt;-<span class="st"> </span><span class="kw">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<ul>
<li>Logtransform data with base 2</li>
</ul>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>pe &lt;-<span class="st"> </span><span class="kw">logTransform</span>(pe, <span class="dt">base =</span> <span class="dv">2</span>, <span class="dt">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Filtering</li>
</ol>
<details>
<summary> Click to see code to filter the data </summary>
<p>
<ul>
<li>Handling overlapping protein groups</li>
</ul>
<p>In our approach a peptide can map to multiple proteins, as long as there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe, <span class="op">~</span><span class="st"> </span>Proteins <span class="op">%in%</span><span class="st"> </span><span class="kw">smallestUniqueGroups</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>Proteins))</span></code></pre></div>
<ul>
<li>Remove reverse sequences (decoys) and contaminants</li>
</ul>
<p>We now remove the contaminants, peptides that map to decoy sequences, and proteins which were only identified by peptides with modifications.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span>Reverse <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span>
<span id="cb36-2"><a href="#cb36-2"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span><span class="st"> </span>Contaminant <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span></code></pre></div>
<ul>
<li>Drop peptides that were only identified in one sample</li>
</ul>
<p>We keep peptides that were observed at last twice.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span><span class="st"> </span>nNonZero <span class="op">&gt;=</span><span class="dv">2</span>)</span>
<span id="cb37-2"><a href="#cb37-2"></a><span class="kw">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 25452</code></pre>
<p>We keep 25452 peptides upon filtering.</p>
<ol start="3" style="list-style-type: decimal">
<li>Normalization</li>
</ol>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a>pe &lt;-<span class="st"> </span><span class="kw">normalize</span>(pe, </span>
<span id="cb39-2"><a href="#cb39-2"></a>                <span class="dt">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb39-3"><a href="#cb39-3"></a>                <span class="dt">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb39-4"><a href="#cb39-4"></a>                <span class="dt">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Summarization</li>
</ol>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a>pe &lt;-<span class="st"> </span><span class="kw">aggregateFeatures</span>(pe,</span>
<span id="cb40-2"><a href="#cb40-2"></a>  <span class="dt">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb40-3"><a href="#cb40-3"></a>  <span class="dt">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb40-4"><a href="#cb40-4"></a>  <span class="dt">na.rm =</span> <span class="ot">TRUE</span>,</span>
<span id="cb40-5"><a href="#cb40-5"></a>  <span class="dt">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="statistical-inference-1" class="section level3">
<h3><span class="header-section-number">2.5.4</span> Statistical Inference</h3>
<details>
<summary> Click to see code to inference </summary>
<p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a>pe &lt;-<span class="st"> </span><span class="kw">msqrob</span>(<span class="dt">object =</span> pe, <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="dt">formula =</span> <span class="op">~</span>prognosis)</span>
<span id="cb41-2"><a href="#cb41-2"></a>L &lt;-<span class="st"> </span><span class="kw">makeContrast</span>(<span class="st">&quot;prognosisPD=0&quot;</span>, <span class="dt">parameterNames =</span> <span class="kw">c</span>(<span class="st">&quot;prognosisPD&quot;</span>))</span>
<span id="cb41-3"><a href="#cb41-3"></a>pe &lt;-<span class="st"> </span><span class="kw">hypothesisTest</span>(<span class="dt">object =</span> pe, <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="dt">contrast =</span> L)</span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a>volcano6x6 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>prognosisPD <span class="op">%&gt;%</span><span class="st"> </span>na.exclude,</span>
<span id="cb42-2"><a href="#cb42-2"></a>                  <span class="kw">aes</span>(<span class="dt">x =</span> logFC, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(pval), <span class="dt">color =</span> adjPval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)) <span class="op">+</span></span>
<span id="cb42-3"><a href="#cb42-3"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">cex =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb42-4"><a href="#cb42-4"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">alpha</span>(<span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb42-5"><a href="#cb42-5"></a><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></span>
<span id="cb42-6"><a href="#cb42-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="kw">sum</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>prognosisPD<span class="op">$</span>adjPval<span class="op">&lt;</span><span class="fl">0.05</span>,<span class="dt">na.rm=</span><span class="ot">TRUE</span>),<span class="st">&quot; proteins are found to be DA&quot;</span>))</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a>volcano6x6</span></code></pre></div>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
</div>
</div>
<div id="experiment-with-a-9-vs-9-comparison" class="section level2">
<h2><span class="header-section-number">2.6</span> Experiment with a 9 vs 9 comparison</h2>
<div id="read-data-2" class="section level3">
<h3><span class="header-section-number">2.6.1</span> Read data</h3>
<details>
<summary> Click to see background and code </summary>
<p>
<ol style="list-style-type: decimal">
<li>We use a peptides.txt file from MS-data quantified with maxquant that contains MS1 intensities summarized at the peptide level.</li>
</ol>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a>peptidesFile &lt;-<span class="st"> &quot;https://raw.githubusercontent.com/statOmics/pda21/data/quantification/cancer/peptides9vs9.txt&quot;</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below:</li>
</ol>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a>ecols &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;Intensity</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="kw">names</span>(<span class="kw">read.delim</span>(peptidesFile)))</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Read the data and store it in QFeatures object</li>
</ol>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a>pe &lt;-<span class="st"> </span><span class="kw">readQFeatures</span>(</span>
<span id="cb46-2"><a href="#cb46-2"></a>  <span class="dt">table =</span> peptidesFile,</span>
<span id="cb46-3"><a href="#cb46-3"></a>  <span class="dt">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb46-4"><a href="#cb46-4"></a>  <span class="dt">ecol =</span> ecols,</span>
<span id="cb46-5"><a href="#cb46-5"></a>  <span class="dt">name =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="design-2" class="section level3">
<h3><span class="header-section-number">2.6.2</span> Design</h3>
<details>
<summary> Click to see background and code </summary>
<p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a>pe <span class="op">%&gt;%</span><span class="st"> </span>colnames</span></code></pre></div>
<pre><code>## CharacterList of length 1
## [[&quot;peptideRaw&quot;]] Intensity.OR.01 Intensity.OR.04 ... Intensity.PD.11</code></pre>
<ul>
<li>Note, that the sample names the outcome.</li>
<li>We update the colData with information on the design</li>
</ul>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a><span class="kw">colData</span>(pe)<span class="op">$</span>prognosis &lt;-<span class="st">  </span>pe[[<span class="st">&quot;peptideRaw&quot;</span>]] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb49-2"><a href="#cb49-2"></a><span class="st">  </span>colnames <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb49-3"><a href="#cb49-3"></a><span class="st">  </span><span class="kw">substr</span>(<span class="dv">11</span>,<span class="dv">12</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb49-4"><a href="#cb49-4"></a><span class="st">  </span>as.factor</span></code></pre></div>
<ul>
<li>We explore the colData</li>
</ul>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a><span class="kw">colData</span>(pe)</span></code></pre></div>
<pre><code>## DataFrame with 18 rows and 1 column
##                 prognosis
##                  &lt;factor&gt;
## Intensity.OR.01        OR
## Intensity.OR.04        OR
## Intensity.OR.07        OR
## Intensity.OR.09        OR
## Intensity.OR.10        OR
## ...                   ...
## Intensity.PD.07        PD
## Intensity.PD.08        PD
## Intensity.PD.09        PD
## Intensity.PD.10        PD
## Intensity.PD.11        PD</code></pre>
</p>
</details>
</div>
<div id="preprocessing-2" class="section level3">
<h3><span class="header-section-number">2.6.3</span> Preprocessing</h3>
<details>
<summary> Click to see code for preprocessing </summary>
<p>
<ol style="list-style-type: decimal">
<li>Log-transform</li>
</ol>
<ul>
<li>We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.</li>
</ul>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a><span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="op">$</span>nNonZero &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</span></code></pre></div>
<ul>
<li>Peptides with zero intensities are missing peptides and should be represent with a <code>NA</code> value rather than <code>0</code>.</li>
</ul>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a>pe &lt;-<span class="st"> </span><span class="kw">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<ul>
<li>Logtransform data with base 2</li>
</ul>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a>pe &lt;-<span class="st"> </span><span class="kw">logTransform</span>(pe, <span class="dt">base =</span> <span class="dv">2</span>, <span class="dt">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Filtering</li>
</ol>
<details>
<summary> Click to see code to filter the data </summary>
<p>
<ul>
<li>Handling overlapping protein groups</li>
</ul>
<p>In our approach a peptide can map to multiple proteins, as long as there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe, <span class="op">~</span><span class="st"> </span>Proteins <span class="op">%in%</span><span class="st"> </span><span class="kw">smallestUniqueGroups</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>Proteins))</span></code></pre></div>
<ul>
<li>Remove reverse sequences (decoys) and contaminants</li>
</ul>
<p>We now remove the contaminants, peptides that map to decoy sequences, and proteins which were only identified by peptides with modifications.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span>Reverse <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span>
<span id="cb56-2"><a href="#cb56-2"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span><span class="st"> </span>Contaminant <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span></code></pre></div>
<ul>
<li>Drop peptides that were only identified in one sample</li>
</ul>
<p>We keep peptides that were observed at last twice.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span><span class="st"> </span>nNonZero <span class="op">&gt;=</span><span class="dv">2</span>)</span>
<span id="cb57-2"><a href="#cb57-2"></a><span class="kw">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 26696</code></pre>
<p>We keep 26696 peptides upon filtering.</p>
<ol start="3" style="list-style-type: decimal">
<li>Normalization</li>
</ol>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1"></a>pe &lt;-<span class="st"> </span><span class="kw">normalize</span>(pe, </span>
<span id="cb59-2"><a href="#cb59-2"></a>                <span class="dt">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb59-3"><a href="#cb59-3"></a>                <span class="dt">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb59-4"><a href="#cb59-4"></a>                <span class="dt">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Summarization</li>
</ol>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1"></a>pe &lt;-<span class="st"> </span><span class="kw">aggregateFeatures</span>(pe,</span>
<span id="cb60-2"><a href="#cb60-2"></a>  <span class="dt">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb60-3"><a href="#cb60-3"></a>  <span class="dt">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb60-4"><a href="#cb60-4"></a>  <span class="dt">na.rm =</span> <span class="ot">TRUE</span>,</span>
<span id="cb60-5"><a href="#cb60-5"></a>  <span class="dt">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="statistical-inference-2" class="section level3">
<h3><span class="header-section-number">2.6.4</span> Statistical Inference</h3>
<details>
<summary> Click to see code to inference </summary>
<p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1"></a>pe &lt;-<span class="st"> </span><span class="kw">msqrob</span>(<span class="dt">object =</span> pe, <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="dt">formula =</span> <span class="op">~</span>prognosis)</span>
<span id="cb61-2"><a href="#cb61-2"></a>L &lt;-<span class="st"> </span><span class="kw">makeContrast</span>(<span class="st">&quot;prognosisPD=0&quot;</span>, <span class="dt">parameterNames =</span> <span class="kw">c</span>(<span class="st">&quot;prognosisPD&quot;</span>))</span>
<span id="cb61-3"><a href="#cb61-3"></a>pe &lt;-<span class="st"> </span><span class="kw">hypothesisTest</span>(<span class="dt">object =</span> pe, <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="dt">contrast =</span> L)</span></code></pre></div>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1"></a>volcano9x9 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>prognosisPD <span class="op">%&gt;%</span><span class="st"> </span>na.exclude,</span>
<span id="cb62-2"><a href="#cb62-2"></a>                  <span class="kw">aes</span>(<span class="dt">x =</span> logFC, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(pval), <span class="dt">color =</span> adjPval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)) <span class="op">+</span></span>
<span id="cb62-3"><a href="#cb62-3"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">cex =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb62-4"><a href="#cb62-4"></a><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">alpha</span>(<span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb62-5"><a href="#cb62-5"></a><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></span>
<span id="cb62-6"><a href="#cb62-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="kw">sum</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>prognosisPD<span class="op">$</span>adjPval<span class="op">&lt;</span><span class="fl">0.05</span>,<span class="dt">na.rm=</span><span class="ot">TRUE</span>),<span class="st">&quot; proteins are found to be DA&quot;</span>))</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1"></a>volcano9x9</span></code></pre></div>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-53-1.png" width="672" /></p>
<ul>
<li><p>We have seen that the sample size is key to recover DA proteins</p></li>
<li><p>Indeed, if a protein is differentially expressed, the value of T-test depends on the effect size, the variability of the protein expression values and the sample size.</p></li>
</ul>
<p><span class="math display">\[
T_g=\frac{\log_2 \text{FC}}{\text{se}_{\log_2 \text{FC}}}
\]</span></p>
<p><span class="math display">\[
T_g=\frac{\widehat{\text{signal}}}{\widehat{\text{Noise}}}
\]</span></p>
<p>For a two group comparison the standard error on the fold change equals</p>
<p><span class="math display">\[
\text{se}_{\log_2 \text{FC}}=\text{SD}\sqrt{\frac{1}{n_1}+\frac{1}{n_2}}
\]</span></p>
<p><span class="math inline">\(\rightarrow\)</span> if number of bio-repeats increases we have a higher power!</p>
</div>
</div>
</div>
<div id="randomized-complete-block-designs" class="section level1">
<h1><span class="header-section-number">3</span> Randomized complete block designs</h1>
<p><span class="math display">\[\sigma^2= \sigma^2_{bio}+\sigma^2_\text{lab} +\sigma^2_\text{extraction} + \sigma^2_\text{run} + \ldots\]</span></p>
<ul>
<li>Biological: fluctuations in protein level between mice, fluctations in protein level between cells, …</li>
<li>Technical: cage effect, lab effect, week effect, plasma extraction, MS-run, …</li>
</ul>
<div id="nature-methods-points-of-significance---blocking" class="section level2">
<h2><span class="header-section-number">3.1</span> Nature methods: Points of significance - Blocking</h2>
<p><a href="https://www.nature.com/articles/nmeth.3005.pdf">https://www.nature.com/articles/nmeth.3005.pdf</a></p>
</div>
<div id="mouse-example" class="section level2">
<h2><span class="header-section-number">3.2</span> Mouse example</h2>
<p><img src="figures/mouseTcell_RCB_design.png" width="50%" /></p>
<p>Duguet et al. (2017) MCP 16(8):1416-1432. doi: 10.1074/mcp.m116.062745</p>
<ul>
<li>All treatments of interest are present within block!</li>
<li>We can estimate the effect of the treatment within block!</li>
</ul>
<p>To illustrate the power of blocking we have subsetted the data of Duguet et al. in a</p>
<ul>
<li><p>completely randomized design with</p>
<ul>
<li>four mice for which we only have measurements on the ordinary T-cells</li>
<li>four mice for which we only have measurements on the regulatory T-cells</li>
</ul></li>
<li><p>randomized complete block design with four mice for which we both have</p>
<ul>
<li>measurements on ordinary T-cells as well as</li>
<li>measurements on regulatory T-cells</li>
</ul></li>
</ul>
<div id="data" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Data</h3>
<details>
<summary> Click to see code </summary>
<p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb64-2"><a href="#cb64-2"></a><span class="kw">library</span>(limma)</span>
<span id="cb64-3"><a href="#cb64-3"></a><span class="kw">library</span>(QFeatures)</span>
<span id="cb64-4"><a href="#cb64-4"></a><span class="kw">library</span>(msqrob2)</span>
<span id="cb64-5"><a href="#cb64-5"></a><span class="kw">library</span>(plotly)</span>
<span id="cb64-6"><a href="#cb64-6"></a><span class="kw">library</span>(gridExtra)</span>
<span id="cb64-7"><a href="#cb64-7"></a></span>
<span id="cb64-8"><a href="#cb64-8"></a>peptidesFile &lt;-<span class="st"> &quot;https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptidesRCB.txt&quot;</span></span>
<span id="cb64-9"><a href="#cb64-9"></a>peptidesFile2 &lt;-<span class="st"> &quot;https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptidesCRD.txt&quot;</span></span>
<span id="cb64-10"><a href="#cb64-10"></a>peptidesFile3 &lt;-<span class="st"> &quot;https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptides.txt&quot;</span></span>
<span id="cb64-11"><a href="#cb64-11"></a></span>
<span id="cb64-12"><a href="#cb64-12"></a>ecols &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;Intensity</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="kw">names</span>(<span class="kw">read.delim</span>(peptidesFile)))</span>
<span id="cb64-13"><a href="#cb64-13"></a>pe &lt;-<span class="st"> </span><span class="kw">readQFeatures</span>(</span>
<span id="cb64-14"><a href="#cb64-14"></a>  <span class="dt">table =</span> peptidesFile,</span>
<span id="cb64-15"><a href="#cb64-15"></a>  <span class="dt">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb64-16"><a href="#cb64-16"></a>  <span class="dt">ecol =</span> ecols,</span>
<span id="cb64-17"><a href="#cb64-17"></a>  <span class="dt">name =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb64-18"><a href="#cb64-18"></a></span>
<span id="cb64-19"><a href="#cb64-19"></a>ecols2 &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;Intensity</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="kw">names</span>(<span class="kw">read.delim</span>(peptidesFile2)))</span>
<span id="cb64-20"><a href="#cb64-20"></a>pe2 &lt;-<span class="st"> </span><span class="kw">readQFeatures</span>(</span>
<span id="cb64-21"><a href="#cb64-21"></a>  <span class="dt">table =</span> peptidesFile2,</span>
<span id="cb64-22"><a href="#cb64-22"></a>  <span class="dt">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb64-23"><a href="#cb64-23"></a>  <span class="dt">ecol =</span> ecols2,</span>
<span id="cb64-24"><a href="#cb64-24"></a>  <span class="dt">name =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb64-25"><a href="#cb64-25"></a></span>
<span id="cb64-26"><a href="#cb64-26"></a>ecols3 &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;Intensity</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="kw">names</span>(<span class="kw">read.delim</span>(peptidesFile3)))</span>
<span id="cb64-27"><a href="#cb64-27"></a>pe3 &lt;-<span class="st"> </span><span class="kw">readQFeatures</span>(</span>
<span id="cb64-28"><a href="#cb64-28"></a>  <span class="dt">table =</span> peptidesFile3,</span>
<span id="cb64-29"><a href="#cb64-29"></a>  <span class="dt">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb64-30"><a href="#cb64-30"></a>  <span class="dt">ecol =</span> ecols3,</span>
<span id="cb64-31"><a href="#cb64-31"></a>  <span class="dt">name =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb64-32"><a href="#cb64-32"></a></span>
<span id="cb64-33"><a href="#cb64-33"></a><span class="co">### Design</span></span>
<span id="cb64-34"><a href="#cb64-34"></a><span class="kw">colData</span>(pe)<span class="op">$</span>celltype &lt;-<span class="st"> </span><span class="kw">substr</span>(</span>
<span id="cb64-35"><a href="#cb64-35"></a>  <span class="kw">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb64-36"><a href="#cb64-36"></a>  <span class="dv">11</span>,</span>
<span id="cb64-37"><a href="#cb64-37"></a>  <span class="dv">14</span>) <span class="op">%&gt;%</span></span>
<span id="cb64-38"><a href="#cb64-38"></a><span class="st">  </span>unlist <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb64-39"><a href="#cb64-39"></a><span class="st">  </span>as.factor</span>
<span id="cb64-40"><a href="#cb64-40"></a></span>
<span id="cb64-41"><a href="#cb64-41"></a><span class="kw">colData</span>(pe)<span class="op">$</span>mouse &lt;-<span class="st"> </span>pe[[<span class="dv">1</span>]] <span class="op">%&gt;%</span></span>
<span id="cb64-42"><a href="#cb64-42"></a><span class="st">  </span>colnames <span class="op">%&gt;%</span></span>
<span id="cb64-43"><a href="#cb64-43"></a><span class="st">  </span><span class="kw">strsplit</span>(<span class="dt">split=</span><span class="st">&quot;[.]&quot;</span>)  <span class="op">%&gt;%</span></span>
<span id="cb64-44"><a href="#cb64-44"></a><span class="st">  </span><span class="kw">sapply</span>(<span class="cf">function</span>(x) x[<span class="dv">3</span>]) <span class="op">%&gt;%</span></span>
<span id="cb64-45"><a href="#cb64-45"></a><span class="st">  </span>as.factor</span>
<span id="cb64-46"><a href="#cb64-46"></a></span>
<span id="cb64-47"><a href="#cb64-47"></a><span class="kw">colData</span>(pe2)<span class="op">$</span>celltype &lt;-<span class="st"> </span><span class="kw">substr</span>(</span>
<span id="cb64-48"><a href="#cb64-48"></a>  <span class="kw">colnames</span>(pe2[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb64-49"><a href="#cb64-49"></a>  <span class="dv">11</span>,</span>
<span id="cb64-50"><a href="#cb64-50"></a>  <span class="dv">14</span>) <span class="op">%&gt;%</span></span>
<span id="cb64-51"><a href="#cb64-51"></a><span class="st">  </span>unlist <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb64-52"><a href="#cb64-52"></a><span class="st">  </span>as.factor</span>
<span id="cb64-53"><a href="#cb64-53"></a></span>
<span id="cb64-54"><a href="#cb64-54"></a><span class="kw">colData</span>(pe2)<span class="op">$</span>mouse &lt;-<span class="st"> </span>pe2[[<span class="dv">1</span>]] <span class="op">%&gt;%</span></span>
<span id="cb64-55"><a href="#cb64-55"></a><span class="st">  </span>colnames <span class="op">%&gt;%</span></span>
<span id="cb64-56"><a href="#cb64-56"></a><span class="st">  </span><span class="kw">strsplit</span>(<span class="dt">split=</span><span class="st">&quot;[.]&quot;</span>)  <span class="op">%&gt;%</span></span>
<span id="cb64-57"><a href="#cb64-57"></a><span class="st">  </span><span class="kw">sapply</span>(<span class="cf">function</span>(x) x[<span class="dv">3</span>]) <span class="op">%&gt;%</span></span>
<span id="cb64-58"><a href="#cb64-58"></a><span class="st">  </span>as.factor</span>
<span id="cb64-59"><a href="#cb64-59"></a></span>
<span id="cb64-60"><a href="#cb64-60"></a><span class="kw">colData</span>(pe3)<span class="op">$</span>celltype &lt;-<span class="st"> </span><span class="kw">substr</span>(</span>
<span id="cb64-61"><a href="#cb64-61"></a>  <span class="kw">colnames</span>(pe3[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb64-62"><a href="#cb64-62"></a>  <span class="dv">11</span>,</span>
<span id="cb64-63"><a href="#cb64-63"></a>  <span class="dv">14</span>) <span class="op">%&gt;%</span></span>
<span id="cb64-64"><a href="#cb64-64"></a><span class="st">  </span>unlist <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb64-65"><a href="#cb64-65"></a><span class="st">  </span>as.factor</span>
<span id="cb64-66"><a href="#cb64-66"></a></span>
<span id="cb64-67"><a href="#cb64-67"></a><span class="kw">colData</span>(pe3)<span class="op">$</span>mouse &lt;-<span class="st"> </span>pe3[[<span class="dv">1</span>]] <span class="op">%&gt;%</span></span>
<span id="cb64-68"><a href="#cb64-68"></a><span class="st">  </span>colnames <span class="op">%&gt;%</span></span>
<span id="cb64-69"><a href="#cb64-69"></a><span class="st">  </span><span class="kw">strsplit</span>(<span class="dt">split=</span><span class="st">&quot;[.]&quot;</span>)  <span class="op">%&gt;%</span></span>
<span id="cb64-70"><a href="#cb64-70"></a><span class="st">  </span><span class="kw">sapply</span>(<span class="cf">function</span>(x) x[<span class="dv">3</span>]) <span class="op">%&gt;%</span></span>
<span id="cb64-71"><a href="#cb64-71"></a><span class="st">  </span>as.factor</span></code></pre></div>
</p>
<p></design></p>
</div>
<div id="preprocessing-3" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Preprocessing</h3>
<div id="log-transform" class="section level4">
<h4><span class="header-section-number">3.2.2.1</span> Log-transform</h4>
<details>
<summary> Click to see code to log-transfrom the data </summary>
<p>
<ul>
<li>We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.</li>
</ul>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1"></a><span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="op">$</span>nNonZero &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</span>
<span id="cb65-2"><a href="#cb65-2"></a></span>
<span id="cb65-3"><a href="#cb65-3"></a><span class="kw">rowData</span>(pe2[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="op">$</span>nNonZero &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">assay</span>(pe2[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</span>
<span id="cb65-4"><a href="#cb65-4"></a></span>
<span id="cb65-5"><a href="#cb65-5"></a><span class="kw">rowData</span>(pe3[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="op">$</span>nNonZero &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">assay</span>(pe3[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</span></code></pre></div>
<ul>
<li>Peptides with zero intensities are missing peptides and should be represent with a <code>NA</code> value rather than <code>0</code>.</li>
</ul>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1"></a>pe &lt;-<span class="st"> </span><span class="kw">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span>
<span id="cb66-2"><a href="#cb66-2"></a></span>
<span id="cb66-3"><a href="#cb66-3"></a>pe2 &lt;-<span class="st"> </span><span class="kw">zeroIsNA</span>(pe2, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span>
<span id="cb66-4"><a href="#cb66-4"></a></span>
<span id="cb66-5"><a href="#cb66-5"></a>pe3 &lt;-<span class="st"> </span><span class="kw">zeroIsNA</span>(pe3, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<ul>
<li>Logtransform data with base 2</li>
</ul>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1"></a>pe &lt;-<span class="st"> </span><span class="kw">logTransform</span>(pe, <span class="dt">base =</span> <span class="dv">2</span>, <span class="dt">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb67-2"><a href="#cb67-2"></a></span>
<span id="cb67-3"><a href="#cb67-3"></a>pe2 &lt;-<span class="st"> </span><span class="kw">logTransform</span>(pe2, <span class="dt">base =</span> <span class="dv">2</span>, <span class="dt">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb67-4"><a href="#cb67-4"></a></span>
<span id="cb67-5"><a href="#cb67-5"></a>pe3 &lt;-<span class="st"> </span><span class="kw">logTransform</span>(pe3, <span class="dt">base =</span> <span class="dv">2</span>, <span class="dt">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="filtering" class="section level4">
<h4><span class="header-section-number">3.2.2.2</span> Filtering</h4>
<details>
<summary> Click to see details on filtering </summary>
<p>
<ol style="list-style-type: decimal">
<li>Handling overlapping protein groups</li>
</ol>
<p>In our approach a peptide can map to multiple proteins, as long as there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe, <span class="op">~</span><span class="st"> </span>Proteins <span class="op">%in%</span><span class="st"> </span><span class="kw">smallestUniqueGroups</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>Proteins))</span>
<span id="cb68-2"><a href="#cb68-2"></a></span>
<span id="cb68-3"><a href="#cb68-3"></a>pe2 &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe2, <span class="op">~</span><span class="st"> </span>Proteins <span class="op">%in%</span><span class="st"> </span><span class="kw">smallestUniqueGroups</span>(<span class="kw">rowData</span>(pe2[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>Proteins))</span>
<span id="cb68-4"><a href="#cb68-4"></a></span>
<span id="cb68-5"><a href="#cb68-5"></a>pe3 &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe3, <span class="op">~</span><span class="st"> </span>Proteins <span class="op">%in%</span><span class="st"> </span><span class="kw">smallestUniqueGroups</span>(<span class="kw">rowData</span>(pe3[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>Proteins))</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Remove reverse sequences (decoys) and contaminants</li>
</ol>
<p>We now remove the contaminants, peptides that map to decoy sequences, and proteins which were only identified by peptides with modifications.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span>Reverse <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span>
<span id="cb69-2"><a href="#cb69-2"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span><span class="st"> </span>Potential.contaminant <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span>
<span id="cb69-3"><a href="#cb69-3"></a></span>
<span id="cb69-4"><a href="#cb69-4"></a>pe2 &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe2,<span class="op">~</span>Reverse <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span>
<span id="cb69-5"><a href="#cb69-5"></a>pe2 &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe2,<span class="op">~</span><span class="st"> </span>Potential.contaminant <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span>
<span id="cb69-6"><a href="#cb69-6"></a></span>
<span id="cb69-7"><a href="#cb69-7"></a>pe3 &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe3,<span class="op">~</span>Reverse <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span>
<span id="cb69-8"><a href="#cb69-8"></a>pe3 &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe3,<span class="op">~</span><span class="st"> </span>Potential.contaminant <span class="op">!=</span><span class="st"> &quot;+&quot;</span>)</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Drop peptides that were only identified in one sample</li>
</ol>
<p>We keep peptides that were observed at last twice.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1"></a>pe &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe,<span class="op">~</span><span class="st"> </span>nNonZero <span class="op">&gt;=</span><span class="dv">2</span>)</span>
<span id="cb70-2"><a href="#cb70-2"></a><span class="kw">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 44449</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1"></a>pe2 &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe2,<span class="op">~</span><span class="st"> </span>nNonZero <span class="op">&gt;=</span><span class="dv">2</span>)</span>
<span id="cb72-2"><a href="#cb72-2"></a><span class="kw">nrow</span>(pe2[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 43401</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1"></a>pe3 &lt;-<span class="st"> </span><span class="kw">filterFeatures</span>(pe3,<span class="op">~</span><span class="st"> </span>nNonZero <span class="op">&gt;=</span><span class="dv">2</span>)</span>
<span id="cb74-2"><a href="#cb74-2"></a><span class="kw">nrow</span>(pe3[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 47431</code></pre>
</p>
</details>
</div>
<div id="normalization" class="section level4">
<h4><span class="header-section-number">3.2.2.3</span> Normalization</h4>
<details>
<summary> Click to see code to normalize the data </summary>
<p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1"></a>pe &lt;-<span class="st"> </span><span class="kw">normalize</span>(pe, </span>
<span id="cb76-2"><a href="#cb76-2"></a>                <span class="dt">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb76-3"><a href="#cb76-3"></a>                <span class="dt">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb76-4"><a href="#cb76-4"></a>                <span class="dt">method =</span> <span class="st">&quot;center.median&quot;</span>)</span>
<span id="cb76-5"><a href="#cb76-5"></a></span>
<span id="cb76-6"><a href="#cb76-6"></a>pe2 &lt;-<span class="st"> </span><span class="kw">normalize</span>(pe2, </span>
<span id="cb76-7"><a href="#cb76-7"></a>                <span class="dt">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb76-8"><a href="#cb76-8"></a>                <span class="dt">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb76-9"><a href="#cb76-9"></a>                <span class="dt">method =</span> <span class="st">&quot;center.median&quot;</span>)</span>
<span id="cb76-10"><a href="#cb76-10"></a></span>
<span id="cb76-11"><a href="#cb76-11"></a></span>
<span id="cb76-12"><a href="#cb76-12"></a>pe3 &lt;-<span class="st"> </span><span class="kw">normalize</span>(pe3, </span>
<span id="cb76-13"><a href="#cb76-13"></a>                <span class="dt">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb76-14"><a href="#cb76-14"></a>                <span class="dt">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb76-15"><a href="#cb76-15"></a>                <span class="dt">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="summarization" class="section level4">
<h4><span class="header-section-number">3.2.2.4</span> Summarization</h4>
<details>
<summary> Click to see code to summarize the data </summary>
<p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1"></a>pe &lt;-<span class="st"> </span><span class="kw">aggregateFeatures</span>(pe,</span>
<span id="cb77-2"><a href="#cb77-2"></a> <span class="dt">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb77-3"><a href="#cb77-3"></a> <span class="dt">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb77-4"><a href="#cb77-4"></a> <span class="dt">na.rm =</span> <span class="ot">TRUE</span>,</span>
<span id="cb77-5"><a href="#cb77-5"></a> <span class="dt">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1"></a>pe2 &lt;-<span class="st"> </span><span class="kw">aggregateFeatures</span>(pe2,</span>
<span id="cb79-2"><a href="#cb79-2"></a> <span class="dt">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb79-3"><a href="#cb79-3"></a> <span class="dt">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb79-4"><a href="#cb79-4"></a> <span class="dt">na.rm =</span> <span class="ot">TRUE</span>,</span>
<span id="cb79-5"><a href="#cb79-5"></a> <span class="dt">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1"></a>pe3 &lt;-<span class="st"> </span><span class="kw">aggregateFeatures</span>(pe3,</span>
<span id="cb81-2"><a href="#cb81-2"></a> <span class="dt">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb81-3"><a href="#cb81-3"></a> <span class="dt">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb81-4"><a href="#cb81-4"></a> <span class="dt">na.rm =</span> <span class="ot">TRUE</span>,</span>
<span id="cb81-5"><a href="#cb81-5"></a> <span class="dt">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
</p>
</details>
</div>
</div>
<div id="data-exploration-what-is-impact-of-blocking" class="section level3">
<h3><span class="header-section-number">3.2.3</span> Data Exploration: what is impact of blocking?</h3>
<details>
<summary> Click to see code </summary>
<p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1"></a><span class="kw">levels</span>(<span class="kw">colData</span>(pe3)<span class="op">$</span>mouse) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;m&quot;</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">7</span>)</span>
<span id="cb83-2"><a href="#cb83-2"></a>mdsObj3 &lt;-<span class="st"> </span><span class="kw">plotMDS</span>(<span class="kw">assay</span>(pe3[[<span class="st">&quot;protein&quot;</span>]]), <span class="dt">plot =</span> <span class="ot">FALSE</span>)</span>
<span id="cb83-3"><a href="#cb83-3"></a>mdsOrig &lt;-<span class="st"> </span><span class="kw">colData</span>(pe3) <span class="op">%&gt;%</span></span>
<span id="cb83-4"><a href="#cb83-4"></a><span class="st">  </span>as.data.frame <span class="op">%&gt;%</span></span>
<span id="cb83-5"><a href="#cb83-5"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">mds1 =</span> mdsObj3<span class="op">$</span>x,</span>
<span id="cb83-6"><a href="#cb83-6"></a>         <span class="dt">mds2 =</span> mdsObj3<span class="op">$</span>y,</span>
<span id="cb83-7"><a href="#cb83-7"></a>         <span class="dt">lab =</span> <span class="kw">paste</span>(mouse,celltype,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb83-8"><a href="#cb83-8"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> mds1, <span class="dt">y =</span> mds2, <span class="dt">label =</span> lab, <span class="dt">color =</span> celltype, <span class="dt">group =</span> mouse)) <span class="op">+</span></span>
<span id="cb83-9"><a href="#cb83-9"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb83-10"><a href="#cb83-10"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></span>
<span id="cb83-11"><a href="#cb83-11"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="op">+</span></span>
<span id="cb83-12"><a href="#cb83-12"></a><span class="st">  </span><span class="kw">xlab</span>(</span>
<span id="cb83-13"><a href="#cb83-13"></a>    <span class="kw">paste0</span>(</span>
<span id="cb83-14"><a href="#cb83-14"></a>      mdsObj3<span class="op">$</span>axislabel,</span>
<span id="cb83-15"><a href="#cb83-15"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb83-16"><a href="#cb83-16"></a>      <span class="dv">1</span>, </span>
<span id="cb83-17"><a href="#cb83-17"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb83-18"><a href="#cb83-18"></a>      <span class="kw">paste0</span>(</span>
<span id="cb83-19"><a href="#cb83-19"></a>        <span class="kw">round</span>(mdsObj3<span class="op">$</span>var.explained[<span class="dv">1</span>] <span class="op">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb83-20"><a href="#cb83-20"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb83-21"><a href="#cb83-21"></a>        ),</span>
<span id="cb83-22"><a href="#cb83-22"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb83-23"><a href="#cb83-23"></a>      )</span>
<span id="cb83-24"><a href="#cb83-24"></a>    ) <span class="op">+</span></span>
<span id="cb83-25"><a href="#cb83-25"></a><span class="st">  </span><span class="kw">ylab</span>(</span>
<span id="cb83-26"><a href="#cb83-26"></a>    <span class="kw">paste0</span>(</span>
<span id="cb83-27"><a href="#cb83-27"></a>      mdsObj3<span class="op">$</span>axislabel,</span>
<span id="cb83-28"><a href="#cb83-28"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb83-29"><a href="#cb83-29"></a>      <span class="dv">2</span>, </span>
<span id="cb83-30"><a href="#cb83-30"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb83-31"><a href="#cb83-31"></a>      <span class="kw">paste0</span>(</span>
<span id="cb83-32"><a href="#cb83-32"></a>        <span class="kw">round</span>(mdsObj3<span class="op">$</span>var.explained[<span class="dv">2</span>] <span class="op">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb83-33"><a href="#cb83-33"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb83-34"><a href="#cb83-34"></a>        ),</span>
<span id="cb83-35"><a href="#cb83-35"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb83-36"><a href="#cb83-36"></a>      )</span>
<span id="cb83-37"><a href="#cb83-37"></a>    ) <span class="op">+</span></span>
<span id="cb83-38"><a href="#cb83-38"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Original (RCB)&quot;</span>)</span>
<span id="cb83-39"><a href="#cb83-39"></a></span>
<span id="cb83-40"><a href="#cb83-40"></a><span class="kw">levels</span>(<span class="kw">colData</span>(pe)<span class="op">$</span>mouse) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;m&quot;</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>)</span>
<span id="cb83-41"><a href="#cb83-41"></a>mdsObj &lt;-<span class="st"> </span><span class="kw">plotMDS</span>(<span class="kw">assay</span>(pe[[<span class="st">&quot;protein&quot;</span>]]), <span class="dt">plot =</span> <span class="ot">FALSE</span>)</span>
<span id="cb83-42"><a href="#cb83-42"></a>mdsRCB &lt;-<span class="st"> </span><span class="kw">colData</span>(pe) <span class="op">%&gt;%</span></span>
<span id="cb83-43"><a href="#cb83-43"></a><span class="st">  </span>as.data.frame <span class="op">%&gt;%</span></span>
<span id="cb83-44"><a href="#cb83-44"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">mds1 =</span> mdsObj<span class="op">$</span>x,</span>
<span id="cb83-45"><a href="#cb83-45"></a>         <span class="dt">mds2 =</span> mdsObj<span class="op">$</span>y,</span>
<span id="cb83-46"><a href="#cb83-46"></a>         <span class="dt">lab =</span> <span class="kw">paste</span>(mouse,celltype,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb83-47"><a href="#cb83-47"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> mds1, <span class="dt">y =</span> mds2, <span class="dt">label =</span> lab, <span class="dt">color =</span> celltype, <span class="dt">group =</span> mouse)) <span class="op">+</span></span>
<span id="cb83-48"><a href="#cb83-48"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb83-49"><a href="#cb83-49"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></span>
<span id="cb83-50"><a href="#cb83-50"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="op">+</span></span>
<span id="cb83-51"><a href="#cb83-51"></a><span class="st">  </span><span class="kw">xlab</span>(</span>
<span id="cb83-52"><a href="#cb83-52"></a>    <span class="kw">paste0</span>(</span>
<span id="cb83-53"><a href="#cb83-53"></a>      mdsObj<span class="op">$</span>axislabel,</span>
<span id="cb83-54"><a href="#cb83-54"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb83-55"><a href="#cb83-55"></a>      <span class="dv">1</span>, </span>
<span id="cb83-56"><a href="#cb83-56"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb83-57"><a href="#cb83-57"></a>      <span class="kw">paste0</span>(</span>
<span id="cb83-58"><a href="#cb83-58"></a>        <span class="kw">round</span>(mdsObj<span class="op">$</span>var.explained[<span class="dv">1</span>] <span class="op">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb83-59"><a href="#cb83-59"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb83-60"><a href="#cb83-60"></a>        ),</span>
<span id="cb83-61"><a href="#cb83-61"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb83-62"><a href="#cb83-62"></a>      )</span>
<span id="cb83-63"><a href="#cb83-63"></a>    ) <span class="op">+</span></span>
<span id="cb83-64"><a href="#cb83-64"></a><span class="st">  </span><span class="kw">ylab</span>(</span>
<span id="cb83-65"><a href="#cb83-65"></a>    <span class="kw">paste0</span>(</span>
<span id="cb83-66"><a href="#cb83-66"></a>      mdsObj<span class="op">$</span>axislabel,</span>
<span id="cb83-67"><a href="#cb83-67"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb83-68"><a href="#cb83-68"></a>      <span class="dv">2</span>, </span>
<span id="cb83-69"><a href="#cb83-69"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb83-70"><a href="#cb83-70"></a>      <span class="kw">paste0</span>(</span>
<span id="cb83-71"><a href="#cb83-71"></a>        <span class="kw">round</span>(mdsObj<span class="op">$</span>var.explained[<span class="dv">2</span>] <span class="op">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb83-72"><a href="#cb83-72"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb83-73"><a href="#cb83-73"></a>        ),</span>
<span id="cb83-74"><a href="#cb83-74"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb83-75"><a href="#cb83-75"></a>      )</span>
<span id="cb83-76"><a href="#cb83-76"></a>    ) <span class="op">+</span></span>
<span id="cb83-77"><a href="#cb83-77"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Randomized Complete Block (RCB)&quot;</span>)</span>
<span id="cb83-78"><a href="#cb83-78"></a></span>
<span id="cb83-79"><a href="#cb83-79"></a></span>
<span id="cb83-80"><a href="#cb83-80"></a><span class="kw">levels</span>(<span class="kw">colData</span>(pe2)<span class="op">$</span>mouse) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;m&quot;</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">8</span>)</span>
<span id="cb83-81"><a href="#cb83-81"></a>mdsObj2 &lt;-<span class="st"> </span><span class="kw">plotMDS</span>(<span class="kw">assay</span>(pe2[[<span class="st">&quot;protein&quot;</span>]]), <span class="dt">plot =</span> <span class="ot">FALSE</span>)</span>
<span id="cb83-82"><a href="#cb83-82"></a>mdsCRD &lt;-<span class="st"> </span><span class="kw">colData</span>(pe2) <span class="op">%&gt;%</span></span>
<span id="cb83-83"><a href="#cb83-83"></a><span class="st">  </span>as.data.frame <span class="op">%&gt;%</span></span>
<span id="cb83-84"><a href="#cb83-84"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">mds1 =</span> mdsObj2<span class="op">$</span>x,</span>
<span id="cb83-85"><a href="#cb83-85"></a>         <span class="dt">mds2 =</span> mdsObj2<span class="op">$</span>y,</span>
<span id="cb83-86"><a href="#cb83-86"></a>         <span class="dt">lab =</span> <span class="kw">paste</span>(mouse,celltype,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb83-87"><a href="#cb83-87"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> mds1, <span class="dt">y =</span> mds2, <span class="dt">label =</span> lab, <span class="dt">color =</span> celltype, <span class="dt">group =</span> mouse)) <span class="op">+</span></span>
<span id="cb83-88"><a href="#cb83-88"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb83-89"><a href="#cb83-89"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></span>
<span id="cb83-90"><a href="#cb83-90"></a><span class="st">  </span><span class="kw">xlab</span>(</span>
<span id="cb83-91"><a href="#cb83-91"></a>    <span class="kw">paste0</span>(</span>
<span id="cb83-92"><a href="#cb83-92"></a>      mdsObj<span class="op">$</span>axislabel,</span>
<span id="cb83-93"><a href="#cb83-93"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb83-94"><a href="#cb83-94"></a>      <span class="dv">1</span>, </span>
<span id="cb83-95"><a href="#cb83-95"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb83-96"><a href="#cb83-96"></a>      <span class="kw">paste0</span>(</span>
<span id="cb83-97"><a href="#cb83-97"></a>        <span class="kw">round</span>(mdsObj2<span class="op">$</span>var.explained[<span class="dv">1</span>] <span class="op">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb83-98"><a href="#cb83-98"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb83-99"><a href="#cb83-99"></a>        ),</span>
<span id="cb83-100"><a href="#cb83-100"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb83-101"><a href="#cb83-101"></a>      )</span>
<span id="cb83-102"><a href="#cb83-102"></a>    ) <span class="op">+</span></span>
<span id="cb83-103"><a href="#cb83-103"></a><span class="st">  </span><span class="kw">ylab</span>(</span>
<span id="cb83-104"><a href="#cb83-104"></a>    <span class="kw">paste0</span>(</span>
<span id="cb83-105"><a href="#cb83-105"></a>      mdsObj<span class="op">$</span>axislabel,</span>
<span id="cb83-106"><a href="#cb83-106"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb83-107"><a href="#cb83-107"></a>      <span class="dv">2</span>, </span>
<span id="cb83-108"><a href="#cb83-108"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb83-109"><a href="#cb83-109"></a>      <span class="kw">paste0</span>(</span>
<span id="cb83-110"><a href="#cb83-110"></a>        <span class="kw">round</span>(mdsObj2<span class="op">$</span>var.explained[<span class="dv">2</span>] <span class="op">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb83-111"><a href="#cb83-111"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb83-112"><a href="#cb83-112"></a>        ),</span>
<span id="cb83-113"><a href="#cb83-113"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb83-114"><a href="#cb83-114"></a>      )</span>
<span id="cb83-115"><a href="#cb83-115"></a>    ) <span class="op">+</span></span>
<span id="cb83-116"><a href="#cb83-116"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Completely Randomized Design (CRD)&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1"></a>mdsOrig</span></code></pre></div>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-65-1.png" width="672" /></p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1"></a>mdsRCB</span></code></pre></div>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-65-2.png" width="672" /></p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1"></a>mdsCRD</span></code></pre></div>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-65-3.png" width="672" /></p>
<ul>
<li><p>We observe that the leading fold change is according to mouse</p></li>
<li><p>In the second dimension we see a separation according to cell-type</p></li>
<li><p>With the Randomized Complete Block design (RCB) we can remove the mouse effect from the analysis!</p></li>
<li><p>We can isolate the between block variability from the analysis using linear model: <span class="math display">\[ 
y \sim \text{celltype} + \text{mouse}
\]</span></p></li>
<li><p>Possible in msqrob2 and MSstats but not possible with Perseus!</p></li>
</ul>
</div>
</div>
<div id="modeling-and-inference" class="section level2">
<h2><span class="header-section-number">3.3</span> Modeling and inference</h2>
<div id="rcb-analysis" class="section level3">
<h3><span class="header-section-number">3.3.1</span> RCB analysis</h3>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1"></a>pe &lt;-<span class="st"> </span><span class="kw">msqrob</span>(</span>
<span id="cb87-2"><a href="#cb87-2"></a>  <span class="dt">object =</span> pe,</span>
<span id="cb87-3"><a href="#cb87-3"></a>  <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb87-4"><a href="#cb87-4"></a>  <span class="dt">formula =</span> <span class="op">~</span><span class="st"> </span>celltype <span class="op">+</span><span class="st"> </span>mouse)</span></code></pre></div>
</div>
<div id="rcb-wrong-analysis" class="section level3">
<h3><span class="header-section-number">3.3.2</span> RCB wrong analysis</h3>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1"></a>pe &lt;-<span class="st"> </span><span class="kw">msqrob</span>(</span>
<span id="cb88-2"><a href="#cb88-2"></a>  <span class="dt">object =</span> pe,</span>
<span id="cb88-3"><a href="#cb88-3"></a>  <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb88-4"><a href="#cb88-4"></a>  <span class="dt">formula =</span> <span class="op">~</span><span class="st"> </span>celltype, <span class="dt">modelColumnName =</span> <span class="st">&quot;wrongModel&quot;</span>)</span></code></pre></div>
</div>
<div id="crd-analysis" class="section level3">
<h3><span class="header-section-number">3.3.3</span> CRD analysis</h3>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1"></a>pe2 &lt;-<span class="st"> </span><span class="kw">msqrob</span>(</span>
<span id="cb89-2"><a href="#cb89-2"></a>  <span class="dt">object =</span> pe2,</span>
<span id="cb89-3"><a href="#cb89-3"></a>  <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb89-4"><a href="#cb89-4"></a>  <span class="dt">formula =</span> <span class="op">~</span><span class="st"> </span>celltype)</span></code></pre></div>
</div>
<div id="estimation-effect-size-and-inference" class="section level3">
<h3><span class="header-section-number">3.3.4</span> Estimation, effect size and inference</h3>
<p>Effect size in RCB</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1"></a><span class="kw">library</span>(ExploreModelMatrix)</span>
<span id="cb90-2"><a href="#cb90-2"></a><span class="kw">VisualizeDesign</span>(<span class="kw">colData</span>(pe),<span class="op">~</span><span class="st"> </span>celltype <span class="op">+</span><span class="st"> </span>mouse)<span class="op">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-69-1.png" width="672" /></p>
<p>Effect size in CRD</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1"></a><span class="kw">VisualizeDesign</span>(<span class="kw">colData</span>(pe2),<span class="op">~</span><span class="st"> </span>celltype)<span class="op">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-70-1.png" width="672" /></p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1"></a>L &lt;-<span class="st"> </span><span class="kw">makeContrast</span>(<span class="st">&quot;celltypeTreg = 0&quot;</span>, <span class="dt">parameterNames =</span> <span class="kw">c</span>(<span class="st">&quot;celltypeTreg&quot;</span>))</span>
<span id="cb94-2"><a href="#cb94-2"></a>pe &lt;-<span class="st"> </span><span class="kw">hypothesisTest</span>(<span class="dt">object =</span> pe, <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="dt">contrast =</span> L)</span>
<span id="cb94-3"><a href="#cb94-3"></a>pe &lt;-<span class="st"> </span><span class="kw">hypothesisTest</span>(<span class="dt">object =</span> pe, <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="dt">contrast =</span> L, <span class="dt">modelColumn =</span> <span class="st">&quot;wrongModel&quot;</span>, <span class="dt">resultsColumnNamePrefix=</span><span class="st">&quot;wrong&quot;</span>)</span>
<span id="cb94-4"><a href="#cb94-4"></a>pe2 &lt;-<span class="st"> </span><span class="kw">hypothesisTest</span>(<span class="dt">object =</span> pe2, <span class="dt">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="dt">contrast =</span> L)</span></code></pre></div>
</div>
<div id="comparison-of-results" class="section level3">
<h3><span class="header-section-number">3.3.5</span> Comparison of results</h3>
<details>
<summary> Click to see code </summary>
<p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1"></a>volcanoRCB &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</span>
<span id="cb95-2"><a href="#cb95-2"></a>    <span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>celltypeTreg,</span>
<span id="cb95-3"><a href="#cb95-3"></a>    <span class="kw">aes</span>(<span class="dt">x =</span> logFC, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(pval), <span class="dt">color =</span> adjPval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)</span>
<span id="cb95-4"><a href="#cb95-4"></a>) <span class="op">+</span></span>
<span id="cb95-5"><a href="#cb95-5"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">cex =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb95-6"><a href="#cb95-6"></a><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">alpha</span>(<span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="op">+</span></span>
<span id="cb95-7"><a href="#cb95-7"></a><span class="st">    </span><span class="kw">theme_minimal</span>() <span class="op">+</span></span>
<span id="cb95-8"><a href="#cb95-8"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;RCB: </span><span class="ch">\n</span><span class="st">&quot;</span>, </span>
<span id="cb95-9"><a href="#cb95-9"></a>                <span class="kw">sum</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>celltypeTreg<span class="op">$</span>adjPval<span class="op">&lt;</span><span class="fl">0.05</span>,<span class="dt">na.rm=</span><span class="ot">TRUE</span>),</span>
<span id="cb95-10"><a href="#cb95-10"></a>            <span class="st">&quot; significant&quot;</span>))</span>
<span id="cb95-11"><a href="#cb95-11"></a></span>
<span id="cb95-12"><a href="#cb95-12"></a>volcanoRCBwrong &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</span>
<span id="cb95-13"><a href="#cb95-13"></a>    <span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>wrongcelltypeTreg,</span>
<span id="cb95-14"><a href="#cb95-14"></a>    <span class="kw">aes</span>(<span class="dt">x =</span> logFC, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(pval), <span class="dt">color =</span> adjPval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)</span>
<span id="cb95-15"><a href="#cb95-15"></a>) <span class="op">+</span></span>
<span id="cb95-16"><a href="#cb95-16"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">cex =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb95-17"><a href="#cb95-17"></a><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">alpha</span>(<span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="op">+</span></span>
<span id="cb95-18"><a href="#cb95-18"></a><span class="st">    </span><span class="kw">theme_minimal</span>() <span class="op">+</span></span>
<span id="cb95-19"><a href="#cb95-19"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;RCB wrong: </span><span class="ch">\n</span><span class="st">&quot;</span>, </span>
<span id="cb95-20"><a href="#cb95-20"></a>                <span class="kw">sum</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>wrongcelltypeTreg<span class="op">$</span>adjPval<span class="op">&lt;</span><span class="fl">0.05</span>,<span class="dt">na.rm=</span><span class="ot">TRUE</span>),</span>
<span id="cb95-21"><a href="#cb95-21"></a>            <span class="st">&quot; significant&quot;</span>))</span>
<span id="cb95-22"><a href="#cb95-22"></a></span>
<span id="cb95-23"><a href="#cb95-23"></a>volcanoCRD &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</span>
<span id="cb95-24"><a href="#cb95-24"></a>    <span class="kw">rowData</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>celltypeTreg,</span>
<span id="cb95-25"><a href="#cb95-25"></a>    <span class="kw">aes</span>(<span class="dt">x =</span> logFC, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(pval), <span class="dt">color =</span> adjPval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)</span>
<span id="cb95-26"><a href="#cb95-26"></a>) <span class="op">+</span></span>
<span id="cb95-27"><a href="#cb95-27"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">cex =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb95-28"><a href="#cb95-28"></a><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">alpha</span>(<span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="op">+</span></span>
<span id="cb95-29"><a href="#cb95-29"></a><span class="st">    </span><span class="kw">theme_minimal</span>() <span class="op">+</span></span>
<span id="cb95-30"><a href="#cb95-30"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;CRD: </span><span class="ch">\n</span><span class="st">&quot;</span>, </span>
<span id="cb95-31"><a href="#cb95-31"></a>                <span class="kw">sum</span>(<span class="kw">rowData</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>celltypeTreg<span class="op">$</span>adjPval<span class="op">&lt;</span><span class="fl">0.05</span>,<span class="dt">na.rm=</span><span class="ot">TRUE</span>),</span>
<span id="cb95-32"><a href="#cb95-32"></a>            <span class="st">&quot; significant&quot;</span>))</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1"></a><span class="kw">grid.arrange</span>(volcanoRCB,volcanoCRD, volcanoRCBwrong,<span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 777 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 382 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 262 rows containing missing values (geom_point).</code></pre>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-73-1.png" width="672" /></p>
</div>
<div id="comparison-of-standard-deviation" class="section level3">
<h3><span class="header-section-number">3.3.6</span> Comparison of standard deviation</h3>
<details>
<summary> Click to see code </summary>
<p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1"></a>accessions &lt;-<span class="st"> </span><span class="kw">rownames</span>(pe[[<span class="st">&quot;protein&quot;</span>]])[<span class="kw">rownames</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">%in%</span><span class="kw">rownames</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])]</span>
<span id="cb100-2"><a href="#cb100-2"></a>dat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</span>
<span id="cb100-3"><a href="#cb100-3"></a><span class="dt">sigmaRBC =</span> <span class="kw">sapply</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>msqrobModels[accessions], getSigmaPosterior),</span>
<span id="cb100-4"><a href="#cb100-4"></a><span class="dt">sigmaRBCwrong =</span> <span class="kw">sapply</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>wrongModel[accessions], getSigmaPosterior),</span>
<span id="cb100-5"><a href="#cb100-5"></a>sigmaCRD &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">rowData</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])<span class="op">$</span>msqrobModels[accessions], getSigmaPosterior)</span>
<span id="cb100-6"><a href="#cb100-6"></a>)</span>
<span id="cb100-7"><a href="#cb100-7"></a></span>
<span id="cb100-8"><a href="#cb100-8"></a> plotRBCvsWrong &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> dat, <span class="kw">aes</span>(sigmaRBC, sigmaRBCwrong)) <span class="op">+</span></span>
<span id="cb100-9"><a href="#cb100-9"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">shape =</span> <span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb100-10"><a href="#cb100-10"></a><span class="st">    </span><span class="kw">scale_x_log10</span>() <span class="op">+</span></span>
<span id="cb100-11"><a href="#cb100-11"></a><span class="st">    </span><span class="kw">scale_y_log10</span>() <span class="op">+</span></span>
<span id="cb100-12"><a href="#cb100-12"></a><span class="st">    </span><span class="kw">geom_abline</span>(<span class="dt">intercept=</span><span class="dv">0</span>,<span class="dt">slope=</span><span class="dv">1</span>)</span>
<span id="cb100-13"><a href="#cb100-13"></a>plotCRDvsWrong &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> dat, <span class="kw">aes</span>(sigmaCRD, sigmaRBCwrong)) <span class="op">+</span></span>
<span id="cb100-14"><a href="#cb100-14"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">shape =</span> <span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb100-15"><a href="#cb100-15"></a><span class="st">    </span><span class="kw">scale_x_log10</span>() <span class="op">+</span></span>
<span id="cb100-16"><a href="#cb100-16"></a><span class="st">    </span><span class="kw">scale_y_log10</span>() <span class="op">+</span></span>
<span id="cb100-17"><a href="#cb100-17"></a><span class="st">    </span><span class="kw">geom_abline</span>(<span class="dt">intercept=</span><span class="dv">0</span>,<span class="dt">slope=</span><span class="dv">1</span>)</span>
<span id="cb100-18"><a href="#cb100-18"></a>plotRBCvsCRD &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> dat, <span class="kw">aes</span>(sigmaRBC, sigmaCRD)) <span class="op">+</span></span>
<span id="cb100-19"><a href="#cb100-19"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">shape =</span> <span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb100-20"><a href="#cb100-20"></a><span class="st">    </span><span class="kw">scale_x_log10</span>() <span class="op">+</span></span>
<span id="cb100-21"><a href="#cb100-21"></a><span class="st">    </span><span class="kw">scale_y_log10</span>() <span class="op">+</span></span>
<span id="cb100-22"><a href="#cb100-22"></a><span class="st">    </span><span class="kw">geom_abline</span>(<span class="dt">intercept=</span><span class="dv">0</span>,<span class="dt">slope=</span><span class="dv">1</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1"></a><span class="kw">grid.arrange</span>(</span>
<span id="cb101-2"><a href="#cb101-2"></a>  plotRBCvsWrong,</span>
<span id="cb101-3"><a href="#cb101-3"></a>  plotCRDvsWrong,</span>
<span id="cb101-4"><a href="#cb101-4"></a>  plotRBCvsCRD,</span>
<span id="cb101-5"><a href="#cb101-5"></a>  <span class="dt">nrow=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 730 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 397 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 743 rows containing missing values (geom_point).</code></pre>
<p><img src="experimentalDesign_files/figure-html/unnamed-chunk-75-1.png" width="672" /></p>
<ul>
<li><p>We clearly observe that the standard deviation of the protein expression in the RCB is smaller for the majority of the proteins than that obtained with the CRD</p></li>
<li><p>The standard deviation of the protein expression RCB where we perform a wrong analysis without considering the blocking factor according to mouse is much larger for the marjority of the proteins than that obtained with the correct analysis.</p></li>
<li><p>Indeed, when we ignore the blocking factor in the RCB design we do not remove the variability according to mouse from the analysis and the mouse effect is absorbed in the error term. The standard deviation than becomes very comparable to that observed in the completely randomised design where we could not remove the mouse effect from the analysis.</p></li>
<li><p>Why are some of the standard deviations for the RCB with the correct analysis larger than than of the RCB with the incorrect analysis that ignored the mouse blocking factor?</p></li>
<li><p>Can you think of a reason why it would not be useful to block on a particular factor?</p></li>
</ul>
</div>
</div>
</div>

<div id="rmd-source-code">---
title: " Experimental design concepts for label-free proteomics experiments"
author: "Lieven Clement"
date: "[statOmics](https://statomics.github.io), Ghent University"
output:
    html_document:
      code_download: true
      theme: cosmo
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
    pdf_document:
      toc: true
      number_sections: true
linkcolor: blue
urlcolor: blue
citecolor: blue

bibliography: msqrob2.bib
      
---

<a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a>

This is part of the online course [Experimental Design and Data-Analysis in Label-Free Quantitative LC/MS Proteomics - A Tutorial with msqrob2 (hupo21)](https://statomics.github.io/hupo21/)


<details><summary> Click to see libraries that are loaded </summary><p>
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(QFeatures)
library(msqrob2)
library(plotly)
library(gridExtra)
```
</p></details>


# Overview

- Sample size 
- Randomized Complete Block Designs 

# Sample size 

## Statistical models 
- Upon preprocessing and summarization we model the log2 transformed protein expression ($y_{i}$) values using a linear model 

\[y_{i}= \beta_0 + \beta_1 x_{i,1} + \beta x_{i,2} + ... + \epsilon_i\]

with 

- $\beta_0$ the intercept
- $\beta_j$ the slope for predictor $x_{i,j}$
- $x_{i,j}$ a continuous predictor (such as age) or a dummy variable that can take values of 0 and 1 and that will used for factors (e.g treatment: cancer, normal). 

Example: 

- Estrogen Receptor Positive Breast cancer tissues from from patients treated with tamoxifen upon recurrence have been assessed in a proteomics study. 
- Half of the patients had a good outcome (or) and the other half had a poor outcome (pd). 
- The proteomes have been assessed using an LTQ-Orbitrap and the thermo output .RAW files were searched with MaxQuant (version 1.4.1.2) against the human proteome database (FASTA version 2012-09, human canonical proteome).

We model the data for a single protein using a model:

\[y_{i}= \beta_0 + \beta_{PD} x_{i,PD} + \epsilon_i\]

with 
$x_{i,PD}=\begin{cases}
0& \text{good outcome}\\
1& \text{poor outcome}
\end{cases}$. 


### Read data 

<details><summary> Click to see background and code </summary><p>
1. We use a peptides.txt file from MS-data quantified with maxquant that 
contains MS1 intensities summarized at the peptide level. 
```{r}
peptidesFile <- "https://raw.githubusercontent.com/statOmics/pda21/data/quantification/cancer/peptides3vs3.txt"
```

2. Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below: 

```{r}
ecols <- grep("Intensity\\.", names(read.delim(peptidesFile)))
```

3. Read the data and store it in  QFeatures object 

```{r}
pe <- readQFeatures(
  table = peptidesFile,
  fnames = 1,
  ecol = ecols,
  name = "peptideRaw", sep="\t")
```
</p></details>

### Design

<details><summary> Click to see background and code </summary><p>

```{r} 
pe %>% colnames
```

- Note, that the sample names the outcome. 
- We update the colData with information on the design

```{r}
colData(pe)$prognosis <-  pe[["peptideRaw"]] %>% 
  colnames %>% 
  substr(11,12) %>% 
  as.factor
```

- We explore the colData

```{r}
colData(pe)
```

</p></details>

### Preprocessing


<details><summary> Click to see code for preprocessing </summary><p>

1. Log-transform

- We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.

```{r}
rowData(pe[["peptideRaw"]])$nNonZero <- rowSums(assay(pe[["peptideRaw"]]) > 0)
```


- Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.

```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA
```

- Logtransform data with base 2

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")
```

2. Filtering

<details><summary> Click to see code to filter the data </summary><p>

- Handling overlapping protein groups

In our approach a peptide can map to multiple proteins, as long as there is
none of these proteins present in a smaller subgroup.

```{r}
pe <- filterFeatures(pe, ~ Proteins %in% smallestUniqueGroups(rowData(pe[["peptideLog"]])$Proteins))
```

- Remove reverse sequences (decoys) and contaminants

We now remove the contaminants, peptides that map to decoy sequences, and proteins
which were only identified by peptides with modifications.

```{r}
pe <- filterFeatures(pe,~Reverse != "+")
pe <- filterFeatures(pe,~ Contaminant != "+")
```

- Drop peptides that were only identified in one sample

We keep peptides that were observed at last twice.

```{r}
pe <- filterFeatures(pe,~ nNonZero >=2)
nrow(pe[["peptideLog"]])
```

We keep `r nrow(pe[["peptideLog"]])` peptides upon filtering.

3. Normalization 

```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```

4. Summarization

```{r warning=FALSE,message=FALSE}
pe <- aggregateFeatures(pe,
  i = "peptideNorm",
  fcol = "Proteins",
  na.rm = TRUE,
  name = "protein")
```
</p></details>

## Effect size? 

```{r}
library(ExploreModelMatrix)
VisualizeDesign(colData(pe),designFormula = ~prognosis)$plotlist
```


\[
\begin{array}{rclcl}
E[Y|OR] &=& \beta_0\\
E[Y|PD] &=& \beta_0 + \beta_{PD}\\
\log_2 FC_{PD-OR} &=&  \beta_0 + \beta_{PD} -  \beta_0 &=&\beta_{PD}
\end{array}
\]


## Statistical Inference

We want to find proteins that are differential abundant

$\rightarrow$ use a statistical test 

- $H_0$: $\log_2 FC_{PD-OR} = 0$  or $\beta_{PD} = 0$
- $H_1$: $\log_2 FC_{PD-OR} \neq 0$  or $\beta_{PD} \neq 0$

## Experiment with 3 vs 3 comparison

<details><summary> Click to see code to inference </summary><p>
```{r message=FALSE,warning=FALSE}
pe <- msqrob(object = pe, i = "protein", formula = ~prognosis)
L <- makeContrast("prognosisPD=0", parameterNames = c("prognosisPD"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L)
```

```{r}
volcano3x3 <- ggplot(rowData(pe[["protein"]])$prognosisPD %>% na.exclude,
                  aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
  geom_point(cex = 2.5) +
  scale_color_manual(values = alpha(c("black", "red"), 0.5)) + 
  theme_minimal() +
  ggtitle(paste0(sum(rowData(pe[["protein"]])$prognosisPD$adjPval<0.05,na.rm=TRUE)," proteins are found to be DA"))
```

</p></details>

```{r}
volcano3x3
```

Upon correction for multiple testing with using the false discovery rate (FDR) method no proteins are found to be differentially expressed. 

Note, that you can refresh the concept of multiple testing and FDR [here] (https://statomics.github.io/PDA21/pda_quantification_inference.html#14_Multiple_hypothesis_testing).

## Experiment with a 6 vs 6 comparison

### Read data 

<details><summary> Click to see background and code </summary><p>
1. We use a peptides.txt file from MS-data quantified with maxquant that 
contains MS1 intensities summarized at the peptide level. 
```{r}
peptidesFile <- "https://raw.githubusercontent.com/statOmics/pda21/data/quantification/cancer/peptides6vs6.txt"
```

2. Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below: 

```{r}
ecols <- grep("Intensity\\.", names(read.delim(peptidesFile)))
```

3. Read the data and store it in  QFeatures object 

```{r}
pe <- readQFeatures(
  table = peptidesFile,
  fnames = 1,
  ecol = ecols,
  name = "peptideRaw", sep="\t")
```
</p></details>

### Design

<details><summary> Click to see background and code </summary><p>

```{r} 
pe %>% colnames
```

- Note, that the sample names the outcome. 
- We update the colData with information on the design

```{r}
colData(pe)$prognosis <-  pe[["peptideRaw"]] %>% 
  colnames %>% 
  substr(11,12) %>% 
  as.factor
```

- We explore the colData

```{r}
colData(pe)
```

</p></details>

### Preprocessing


<details><summary> Click to see code for preprocessing </summary><p>

1. Log-transform

- We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.

```{r}
rowData(pe[["peptideRaw"]])$nNonZero <- rowSums(assay(pe[["peptideRaw"]]) > 0)
```


- Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.

```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA
```

- Logtransform data with base 2

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")
```

2. Filtering

<details><summary> Click to see code to filter the data </summary><p>

- Handling overlapping protein groups

In our approach a peptide can map to multiple proteins, as long as there is
none of these proteins present in a smaller subgroup.

```{r}
pe <- filterFeatures(pe, ~ Proteins %in% smallestUniqueGroups(rowData(pe[["peptideLog"]])$Proteins))
```

- Remove reverse sequences (decoys) and contaminants

We now remove the contaminants, peptides that map to decoy sequences, and proteins
which were only identified by peptides with modifications.

```{r}
pe <- filterFeatures(pe,~Reverse != "+")
pe <- filterFeatures(pe,~ Contaminant != "+")
```

- Drop peptides that were only identified in one sample

We keep peptides that were observed at last twice.

```{r}
pe <- filterFeatures(pe,~ nNonZero >=2)
nrow(pe[["peptideLog"]])
```

We keep `r nrow(pe[["peptideLog"]])` peptides upon filtering.

3. Normalization 

```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```

4. Summarization

```{r warning=FALSE,message=FALSE}
pe <- aggregateFeatures(pe,
  i = "peptideNorm",
  fcol = "Proteins",
  na.rm = TRUE,
  name = "protein")
```
</p></details>

### Statistical Inference

<details><summary> Click to see code to inference </summary><p>
```{r message=FALSE,warning=FALSE}
pe <- msqrob(object = pe, i = "protein", formula = ~prognosis)
L <- makeContrast("prognosisPD=0", parameterNames = c("prognosisPD"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L)
```

```{r}
volcano6x6 <- ggplot(rowData(pe[["protein"]])$prognosisPD %>% na.exclude,
                  aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
  geom_point(cex = 2.5) +
  scale_color_manual(values = alpha(c("black", "red"), 0.5)) + 
  theme_minimal() +
  ggtitle(paste0(sum(rowData(pe[["protein"]])$prognosisPD$adjPval<0.05,na.rm=TRUE)," proteins are found to be DA"))
```
</p></details>


```{r}
volcano6x6
```


## Experiment with a 9 vs 9 comparison

### Read data 

<details><summary> Click to see background and code </summary><p>
1. We use a peptides.txt file from MS-data quantified with maxquant that 
contains MS1 intensities summarized at the peptide level. 
```{r}
peptidesFile <- "https://raw.githubusercontent.com/statOmics/pda21/data/quantification/cancer/peptides9vs9.txt"
```

2. Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below: 

```{r}
ecols <- grep("Intensity\\.", names(read.delim(peptidesFile)))
```

3. Read the data and store it in  QFeatures object 

```{r}
pe <- readQFeatures(
  table = peptidesFile,
  fnames = 1,
  ecol = ecols,
  name = "peptideRaw", sep="\t")
```
</p></details>

### Design

<details><summary> Click to see background and code </summary><p>

```{r} 
pe %>% colnames
```

- Note, that the sample names the outcome. 
- We update the colData with information on the design

```{r}
colData(pe)$prognosis <-  pe[["peptideRaw"]] %>% 
  colnames %>% 
  substr(11,12) %>% 
  as.factor
```

- We explore the colData

```{r}
colData(pe)
```

</p></details>

### Preprocessing


<details><summary> Click to see code for preprocessing </summary><p>

1. Log-transform

- We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.

```{r}
rowData(pe[["peptideRaw"]])$nNonZero <- rowSums(assay(pe[["peptideRaw"]]) > 0)
```


- Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.

```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA
```

- Logtransform data with base 2

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")
```

2. Filtering

<details><summary> Click to see code to filter the data </summary><p>

- Handling overlapping protein groups

In our approach a peptide can map to multiple proteins, as long as there is
none of these proteins present in a smaller subgroup.

```{r}
pe <- filterFeatures(pe, ~ Proteins %in% smallestUniqueGroups(rowData(pe[["peptideLog"]])$Proteins))
```

- Remove reverse sequences (decoys) and contaminants

We now remove the contaminants, peptides that map to decoy sequences, and proteins
which were only identified by peptides with modifications.

```{r}
pe <- filterFeatures(pe,~Reverse != "+")
pe <- filterFeatures(pe,~ Contaminant != "+")
```

- Drop peptides that were only identified in one sample

We keep peptides that were observed at last twice.

```{r}
pe <- filterFeatures(pe,~ nNonZero >=2)
nrow(pe[["peptideLog"]])
```

We keep `r nrow(pe[["peptideLog"]])` peptides upon filtering.

3. Normalization 

```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```

4. Summarization

```{r warning=FALSE,message=FALSE}
pe <- aggregateFeatures(pe,
  i = "peptideNorm",
  fcol = "Proteins",
  na.rm = TRUE,
  name = "protein")
```
</p></details>

### Statistical Inference

<details><summary> Click to see code to inference </summary><p>
```{r message=FALSE,warning=FALSE}
pe <- msqrob(object = pe, i = "protein", formula = ~prognosis)
L <- makeContrast("prognosisPD=0", parameterNames = c("prognosisPD"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L)
```

```{r}
volcano9x9 <- ggplot(rowData(pe[["protein"]])$prognosisPD %>% na.exclude,
                  aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
  geom_point(cex = 2.5) +
  scale_color_manual(values = alpha(c("black", "red"), 0.5)) + 
  theme_minimal() +
  ggtitle(paste0(sum(rowData(pe[["protein"]])$prognosisPD$adjPval<0.05,na.rm=TRUE)," proteins are found to be DA"))
```
</p></details>


```{r}
volcano9x9
```

- We have seen that the sample size is key to recover DA proteins

- Indeed, if a protein is differentially expressed, the value of T-test depends on the effect size, the variability of the protein expression values and the sample size. 

$$
T_g=\frac{\log_2 \text{FC}}{\text{se}_{\log_2 \text{FC}}}
$$

$$
T_g=\frac{\widehat{\text{signal}}}{\widehat{\text{Noise}}}
$$

For a two group comparison the standard error on the fold change equals

$$
\text{se}_{\log_2 \text{FC}}=\text{SD}\sqrt{\frac{1}{n_1}+\frac{1}{n_2}}
$$

$\rightarrow$ if number of bio-repeats increases we have a higher power!


# Randomized complete block designs

\[\sigma^2= \sigma^2_{bio}+\sigma^2_\text{lab} +\sigma^2_\text{extraction} + \sigma^2_\text{run} + \ldots\]

- Biological: fluctuations in protein level between mice, fluctations in protein level between cells, ...
- Technical: cage effect, lab effect, week effect, plasma extraction, MS-run, ...

## Nature methods: Points of significance - Blocking 

[https://www.nature.com/articles/nmeth.3005.pdf](https://www.nature.com/articles/nmeth.3005.pdf)


## Mouse example 


```{r echo=FALSE, out.width="50%"}
knitr::include_graphics("./figures/mouseTcell_RCB_design.png")
```

Duguet et al. (2017) MCP 16(8):1416-1432. doi: 10.1074/mcp.m116.062745

- All treatments of interest are present within block!
- We can estimate the effect of the treatment within block!

To illustrate the power of blocking we have subsetted the data of Duguet et al. in a 

- completely randomized design with 
    - four mice for which we only have measurements on the ordinary T-cells 
    - four mice for which we only have measurements on the regulatory T-cells 
    
- randomized complete block design with four mice for which we both have
    
    - measurements on ordinary T-cells as well as
    - measurements on regulatory T-cells




### Data 
<details><summary> Click to see code  </summary><p>
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(QFeatures)
library(msqrob2)
library(plotly)
library(gridExtra)

peptidesFile <- "https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptidesRCB.txt"
peptidesFile2 <- "https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptidesCRD.txt"
peptidesFile3 <- "https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptides.txt"

ecols <- grep("Intensity\\.", names(read.delim(peptidesFile)))
pe <- readQFeatures(
  table = peptidesFile,
  fnames = 1,
  ecol = ecols,
  name = "peptideRaw", sep="\t")

ecols2 <- grep("Intensity\\.", names(read.delim(peptidesFile2)))
pe2 <- readQFeatures(
  table = peptidesFile2,
  fnames = 1,
  ecol = ecols2,
  name = "peptideRaw", sep="\t")

ecols3 <- grep("Intensity\\.", names(read.delim(peptidesFile3)))
pe3 <- readQFeatures(
  table = peptidesFile3,
  fnames = 1,
  ecol = ecols3,
  name = "peptideRaw", sep="\t")

### Design
colData(pe)$celltype <- substr(
  colnames(pe[["peptideRaw"]]),
  11,
  14) %>%
  unlist %>%  
  as.factor

colData(pe)$mouse <- pe[[1]] %>%
  colnames %>%
  strsplit(split="[.]")  %>%
  sapply(function(x) x[3]) %>%
  as.factor

colData(pe2)$celltype <- substr(
  colnames(pe2[["peptideRaw"]]),
  11,
  14) %>%
  unlist %>%  
  as.factor

colData(pe2)$mouse <- pe2[[1]] %>%
  colnames %>%
  strsplit(split="[.]")  %>%
  sapply(function(x) x[3]) %>%
  as.factor

colData(pe3)$celltype <- substr(
  colnames(pe3[["peptideRaw"]]),
  11,
  14) %>%
  unlist %>%  
  as.factor

colData(pe3)$mouse <- pe3[[1]] %>%
  colnames %>%
  strsplit(split="[.]")  %>%
  sapply(function(x) x[3]) %>%
  as.factor
```
</p></design>

### Preprocessing 


#### Log-transform

<details><summary> Click to see code to log-transfrom the data </summary><p>
- We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.

```{r}
rowData(pe[["peptideRaw"]])$nNonZero <- rowSums(assay(pe[["peptideRaw"]]) > 0)

rowData(pe2[["peptideRaw"]])$nNonZero <- rowSums(assay(pe2[["peptideRaw"]]) > 0)

rowData(pe3[["peptideRaw"]])$nNonZero <- rowSums(assay(pe3[["peptideRaw"]]) > 0)
```

- Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.

```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA

pe2 <- zeroIsNA(pe2, "peptideRaw") # convert 0 to NA

pe3 <- zeroIsNA(pe3, "peptideRaw") # convert 0 to NA
```

- Logtransform data with base 2

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")

pe2 <- logTransform(pe2, base = 2, i = "peptideRaw", name = "peptideLog")

pe3 <- logTransform(pe3, base = 2, i = "peptideRaw", name = "peptideLog")
```
</p></details>


#### Filtering
<details><summary> Click to see details on filtering </summary><p>

1. Handling overlapping protein groups

In our approach a peptide can map to multiple proteins, as long as there is
none of these proteins present in a smaller subgroup.

```{r}
pe <- filterFeatures(pe, ~ Proteins %in% smallestUniqueGroups(rowData(pe[["peptideLog"]])$Proteins))

pe2 <- filterFeatures(pe2, ~ Proteins %in% smallestUniqueGroups(rowData(pe2[["peptideLog"]])$Proteins))

pe3 <- filterFeatures(pe3, ~ Proteins %in% smallestUniqueGroups(rowData(pe3[["peptideLog"]])$Proteins))
```
2. Remove reverse sequences (decoys) and contaminants

We now remove the contaminants, peptides that map to decoy sequences, and proteins
which were only identified by peptides with modifications.

```{r}
pe <- filterFeatures(pe,~Reverse != "+")
pe <- filterFeatures(pe,~ Potential.contaminant != "+")

pe2 <- filterFeatures(pe2,~Reverse != "+")
pe2 <- filterFeatures(pe2,~ Potential.contaminant != "+")

pe3 <- filterFeatures(pe3,~Reverse != "+")
pe3 <- filterFeatures(pe3,~ Potential.contaminant != "+")
```


3. Drop peptides that were only identified in one sample

We keep peptides that were observed at last twice.

```{r}
pe <- filterFeatures(pe,~ nNonZero >=2)
nrow(pe[["peptideLog"]])

pe2 <- filterFeatures(pe2,~ nNonZero >=2)
nrow(pe2[["peptideLog"]])

pe3 <- filterFeatures(pe3,~ nNonZero >=2)
nrow(pe3[["peptideLog"]])
```

</p></details>

#### Normalization 

<details><summary> Click to see code to normalize the data </summary><p>
```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")

pe2 <- normalize(pe2, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")


pe3 <- normalize(pe3, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```

</p></details>

#### Summarization

<details><summary> Click to see code to summarize the data </summary><p>

```{r,warning=FALSE}
pe <- aggregateFeatures(pe,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "protein")


pe2 <- aggregateFeatures(pe2,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "protein")

pe3 <- aggregateFeatures(pe3,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "protein")
```

</p></details>

### Data Exploration: what is impact of blocking? 

<details><summary> Click to see code </summary><p>
```{r}
levels(colData(pe3)$mouse) <- paste0("m",1:7)
mdsObj3 <- plotMDS(assay(pe3[["protein"]]), plot = FALSE)
mdsOrig <- colData(pe3) %>%
  as.data.frame %>%
  mutate(mds1 = mdsObj3$x,
         mds2 = mdsObj3$y,
         lab = paste(mouse,celltype,sep="_")) %>%
  ggplot(aes(x = mds1, y = mds2, label = lab, color = celltype, group = mouse)) +
  geom_text(show.legend = FALSE) +
  geom_point(shape = 21) +
  geom_line(color = "black", linetype = "dashed") +
  xlab(
    paste0(
      mdsObj3$axislabel,
      " ",
      1, 
      " (",
      paste0(
        round(mdsObj3$var.explained[1] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ylab(
    paste0(
      mdsObj3$axislabel,
      " ",
      2, 
      " (",
      paste0(
        round(mdsObj3$var.explained[2] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ggtitle("Original (RCB)")

levels(colData(pe)$mouse) <- paste0("m",1:4)
mdsObj <- plotMDS(assay(pe[["protein"]]), plot = FALSE)
mdsRCB <- colData(pe) %>%
  as.data.frame %>%
  mutate(mds1 = mdsObj$x,
         mds2 = mdsObj$y,
         lab = paste(mouse,celltype,sep="_")) %>%
  ggplot(aes(x = mds1, y = mds2, label = lab, color = celltype, group = mouse)) +
  geom_text(show.legend = FALSE) +
  geom_point(shape = 21) +
  geom_line(color = "black", linetype = "dashed") +
  xlab(
    paste0(
      mdsObj$axislabel,
      " ",
      1, 
      " (",
      paste0(
        round(mdsObj$var.explained[1] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ylab(
    paste0(
      mdsObj$axislabel,
      " ",
      2, 
      " (",
      paste0(
        round(mdsObj$var.explained[2] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ggtitle("Randomized Complete Block (RCB)")


levels(colData(pe2)$mouse) <- paste0("m",1:8)
mdsObj2 <- plotMDS(assay(pe2[["protein"]]), plot = FALSE)
mdsCRD <- colData(pe2) %>%
  as.data.frame %>%
  mutate(mds1 = mdsObj2$x,
         mds2 = mdsObj2$y,
         lab = paste(mouse,celltype,sep="_")) %>%
  ggplot(aes(x = mds1, y = mds2, label = lab, color = celltype, group = mouse)) +
  geom_text(show.legend = FALSE) +
  geom_point(shape = 21) +
  xlab(
    paste0(
      mdsObj$axislabel,
      " ",
      1, 
      " (",
      paste0(
        round(mdsObj2$var.explained[1] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ylab(
    paste0(
      mdsObj$axislabel,
      " ",
      2, 
      " (",
      paste0(
        round(mdsObj2$var.explained[2] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ggtitle("Completely Randomized Design (CRD)")
```
</p></details>
```{r}
mdsOrig
mdsRCB
mdsCRD
```

- We observe that the leading fold change is according to mouse
- In the second dimension we see a separation according to cell-type 
- With the Randomized Complete Block design (RCB) we can remove the mouse effect from the analysis!




- We can isolate the between block variability from the analysis using linear model:
$$ 
y \sim \text{celltype} + \text{mouse}
$$

- Possible in msqrob2 and MSstats but not possible with Perseus!



## Modeling and inference

### RCB analysis
```{r warning=FALSE}
pe <- msqrob(
  object = pe,
  i = "protein",
  formula = ~ celltype + mouse)
```

### RCB wrong analysis
```{r warning=FALSE}
pe <- msqrob(
  object = pe,
  i = "protein",
  formula = ~ celltype, modelColumnName = "wrongModel")
```

### CRD analysis 
```{r warning = FALSE}
pe2 <- msqrob(
  object = pe2,
  i = "protein",
  formula = ~ celltype)
```

### Estimation, effect size and inference

Effect size in RCB
```{r}
library(ExploreModelMatrix)
VisualizeDesign(colData(pe),~ celltype + mouse)$plotlist
```

Effect size in CRD
```{r}
VisualizeDesign(colData(pe2),~ celltype)$plotlist
```

```{r}
L <- makeContrast("celltypeTreg = 0", parameterNames = c("celltypeTreg"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L)
pe <- hypothesisTest(object = pe, i = "protein", contrast = L, modelColumn = "wrongModel", resultsColumnNamePrefix="wrong")
pe2 <- hypothesisTest(object = pe2, i = "protein", contrast = L)
```

### Comparison of results

<details><summary> Click to see code </summary><p>
```{r}
volcanoRCB <- ggplot(
    rowData(pe[["protein"]])$celltypeTreg,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("RCB: \n", 
                sum(rowData(pe[["protein"]])$celltypeTreg$adjPval<0.05,na.rm=TRUE),
            " significant"))

volcanoRCBwrong <- ggplot(
    rowData(pe[["protein"]])$wrongcelltypeTreg,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("RCB wrong: \n", 
                sum(rowData(pe[["protein"]])$wrongcelltypeTreg$adjPval<0.05,na.rm=TRUE),
            " significant"))

volcanoCRD <- ggplot(
    rowData(pe2[["protein"]])$celltypeTreg,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("CRD: \n", 
                sum(rowData(pe2[["protein"]])$celltypeTreg$adjPval<0.05,na.rm=TRUE),
            " significant"))
```
</p></details>
  
```{r}
grid.arrange(volcanoRCB,volcanoCRD, volcanoRCBwrong,ncol=2)
```

### Comparison of standard deviation 

<details><summary> Click to see code </summary><p>
```{r}
accessions <- rownames(pe[["protein"]])[rownames(pe[["protein"]])%in%rownames(pe2[["protein"]])]
dat <- data.frame(
sigmaRBC = sapply(rowData(pe[["protein"]])$msqrobModels[accessions], getSigmaPosterior),
sigmaRBCwrong = sapply(rowData(pe[["protein"]])$wrongModel[accessions], getSigmaPosterior),
sigmaCRD <- sapply(rowData(pe2[["protein"]])$msqrobModels[accessions], getSigmaPosterior)
)

 plotRBCvsWrong <- ggplot(data = dat, aes(sigmaRBC, sigmaRBCwrong)) +
    geom_point(alpha = 0.1, shape = 20) +
    scale_x_log10() +
    scale_y_log10() +
    geom_abline(intercept=0,slope=1)
plotCRDvsWrong <- ggplot(data = dat, aes(sigmaCRD, sigmaRBCwrong)) +
    geom_point(alpha = 0.1, shape = 20) +
    scale_x_log10() +
    scale_y_log10() +
    geom_abline(intercept=0,slope=1)
plotRBCvsCRD <- ggplot(data = dat, aes(sigmaRBC, sigmaCRD)) +
    geom_point(alpha = 0.1, shape = 20) +
    scale_x_log10() +
    scale_y_log10() +
    geom_abline(intercept=0,slope=1)
```
</p></details>

```{r}
grid.arrange(
  plotRBCvsWrong,
  plotCRDvsWrong,
  plotRBCvsCRD,
  nrow=2)
```

- We clearly observe that the standard deviation of the protein expression in the RCB is smaller for the majority of the proteins than that obtained with the CRD

- The standard deviation of the protein expression RCB where we perform a wrong analysis without considering the blocking factor according to mouse is much larger for the marjority of the proteins than that obtained with the correct analysis. 

- Indeed, when we ignore the blocking factor in the RCB design we do not remove the variability according to mouse from the analysis and the mouse effect is absorbed in the error term. The standard deviation than becomes very comparable to that observed in the completely randomised design where we could not remove the mouse effect from the analysis. 

- Why are some of the standard deviations for the RCB with the correct analysis larger than than of the RCB with the incorrect analysis that ignored the mouse blocking factor?

- Can you think of a reason why it would not be useful to block on a particular factor? 

</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("experimentalDesign.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
